<!doctype html>
<html lang="en" dir="ltr" class="docs-wrapper docs-doc-page docs-version-current plugin-docs plugin-id-default docs-doc-id-Software/galaxy">
<head>
<meta charset="UTF-8">
<meta name="generator" content="Docusaurus v2.4.0">
<title data-rh="true">Galaxy | Lab Wiki</title><meta data-rh="true" name="viewport" content="width=device-width,initial-scale=1"><meta data-rh="true" name="twitter:card" content="summary_large_image"><meta data-rh="true" property="og:url" content="https://CEGRcode.github.io/labtutorial/Software/galaxy"><meta data-rh="true" name="docusaurus_locale" content="en"><meta data-rh="true" name="docsearch:language" content="en"><meta data-rh="true" name="docusaurus_version" content="current"><meta data-rh="true" name="docusaurus_tag" content="docs-default-current"><meta data-rh="true" name="docsearch:version" content="current"><meta data-rh="true" name="docsearch:docusaurus_tag" content="docs-default-current"><meta data-rh="true" property="og:title" content="Galaxy | Lab Wiki"><meta data-rh="true" name="description" content="Dependencies"><meta data-rh="true" property="og:description" content="Dependencies"><link data-rh="true" rel="icon" href="/labtutorial/img/favicon.png"><link data-rh="true" rel="canonical" href="https://CEGRcode.github.io/labtutorial/Software/galaxy"><link data-rh="true" rel="alternate" href="https://CEGRcode.github.io/labtutorial/Software/galaxy" hreflang="en"><link data-rh="true" rel="alternate" href="https://CEGRcode.github.io/labtutorial/Software/galaxy" hreflang="x-default"><link rel="stylesheet" href="/labtutorial/assets/css/styles.0bb38f29.css">
<link rel="preload" href="/labtutorial/assets/js/runtime~main.0d187c5d.js" as="script">
<link rel="preload" href="/labtutorial/assets/js/main.3364e39b.js" as="script">
</head>
<body class="navigation-with-keyboard">
<script>!function(){function t(t){document.documentElement.setAttribute("data-theme",t)}var e=function(){var t=null;try{t=new URLSearchParams(window.location.search).get("docusaurus-theme")}catch(t){}return t}()||function(){var t=null;try{t=localStorage.getItem("theme")}catch(t){}return t}();t(null!==e?e:"light")}()</script><div id="__docusaurus">
<div role="region" aria-label="Skip to main content"><a class="skipToContent_fXgn" href="#docusaurus_skipToContent_fallback">Skip to main content</a></div><nav aria-label="Main" class="navbar navbar--fixed-top"><div class="navbar__inner"><div class="navbar__items"><button aria-label="Toggle navigation bar" aria-expanded="false" class="navbar__toggle clean-btn" type="button"><svg width="30" height="30" viewBox="0 0 30 30" aria-hidden="true"><path stroke="currentColor" stroke-linecap="round" stroke-miterlimit="10" stroke-width="2" d="M4 7h22M4 15h22M4 23h22"></path></svg></button><a href="http://pughlab.mbg.cornell.edu/" target="_blank" rel="noopener noreferrer" class="navbar__brand"><div class="navbar__logo"><img src="/labtutorial/img/logo.svg" alt="Lab logo" class="themedImage_ToTc themedImage--light_HNdA"><img src="/labtutorial/img/logo_dark.svg" alt="Lab logo" class="themedImage_ToTc themedImage--dark_i4oU"></div></a><a class="navbar__item navbar__link" href="/labtutorial/">Wiki</a><a aria-current="page" class="navbar__item navbar__link navbar__link--active" href="/labtutorial/Software/faq">Software</a><a class="navbar__item navbar__link" href="/labtutorial/WetBench/introduction">Wet Bench</a></div><div class="navbar__items navbar__items--right"><a href="https://github.com/CEGRcode" target="_blank" rel="noopener noreferrer" class="navbar__item navbar__link header-github-link"></a><div class="toggle_vylO colorModeToggle_DEke"><button class="clean-btn toggleButton_gllP toggleButtonDisabled_aARS" type="button" disabled="" title="Switch between dark and light mode (currently light mode)" aria-label="Switch between dark and light mode (currently light mode)" aria-live="polite"><svg viewBox="0 0 24 24" width="24" height="24" class="lightToggleIcon_pyhR"><path fill="currentColor" d="M12,9c1.65,0,3,1.35,3,3s-1.35,3-3,3s-3-1.35-3-3S10.35,9,12,9 M12,7c-2.76,0-5,2.24-5,5s2.24,5,5,5s5-2.24,5-5 S14.76,7,12,7L12,7z M2,13l2,0c0.55,0,1-0.45,1-1s-0.45-1-1-1l-2,0c-0.55,0-1,0.45-1,1S1.45,13,2,13z M20,13l2,0c0.55,0,1-0.45,1-1 s-0.45-1-1-1l-2,0c-0.55,0-1,0.45-1,1S19.45,13,20,13z M11,2v2c0,0.55,0.45,1,1,1s1-0.45,1-1V2c0-0.55-0.45-1-1-1S11,1.45,11,2z M11,20v2c0,0.55,0.45,1,1,1s1-0.45,1-1v-2c0-0.55-0.45-1-1-1C11.45,19,11,19.45,11,20z M5.99,4.58c-0.39-0.39-1.03-0.39-1.41,0 c-0.39,0.39-0.39,1.03,0,1.41l1.06,1.06c0.39,0.39,1.03,0.39,1.41,0s0.39-1.03,0-1.41L5.99,4.58z M18.36,16.95 c-0.39-0.39-1.03-0.39-1.41,0c-0.39,0.39-0.39,1.03,0,1.41l1.06,1.06c0.39,0.39,1.03,0.39,1.41,0c0.39-0.39,0.39-1.03,0-1.41 L18.36,16.95z M19.42,5.99c0.39-0.39,0.39-1.03,0-1.41c-0.39-0.39-1.03-0.39-1.41,0l-1.06,1.06c-0.39,0.39-0.39,1.03,0,1.41 s1.03,0.39,1.41,0L19.42,5.99z M7.05,18.36c0.39-0.39,0.39-1.03,0-1.41c-0.39-0.39-1.03-0.39-1.41,0l-1.06,1.06 c-0.39,0.39-0.39,1.03,0,1.41s1.03,0.39,1.41,0L7.05,18.36z"></path></svg><svg viewBox="0 0 24 24" width="24" height="24" class="darkToggleIcon_wfgR"><path fill="currentColor" d="M9.37,5.51C9.19,6.15,9.1,6.82,9.1,7.5c0,4.08,3.32,7.4,7.4,7.4c0.68,0,1.35-0.09,1.99-0.27C17.45,17.19,14.93,19,12,19 c-3.86,0-7-3.14-7-7C5,9.07,6.81,6.55,9.37,5.51z M12,3c-4.97,0-9,4.03-9,9s4.03,9,9,9s9-4.03,9-9c0-0.46-0.04-0.92-0.1-1.36 c-0.98,1.37-2.58,2.26-4.4,2.26c-2.98,0-5.4-2.42-5.4-5.4c0-1.81,0.89-3.42,2.26-4.4C12.92,3.04,12.46,3,12,3L12,3z"></path></svg></button></div><div class="searchBox_ZlJk"></div></div></div><div role="presentation" class="navbar-sidebar__backdrop"></div></nav><div id="docusaurus_skipToContent_fallback" class="main-wrapper mainWrapper_z2l0 docsWrapper_BCFX"><button aria-label="Scroll back to top" class="clean-btn theme-back-to-top-button backToTopButton_sjWU" type="button"></button><div class="docPage__5DB"><aside class="theme-doc-sidebar-container docSidebarContainer_b6E3"><div class="sidebarViewport_Xe31"><div class="sidebar_njMd"><nav aria-label="Docs sidebar" class="menu thin-scrollbar menu_SIkG"><ul class="theme-doc-sidebar-menu menu__list"><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-1 menu__list-item"><a class="menu__link menu__link--active" aria-current="page" href="/labtutorial/Software/galaxy">Galaxy</a></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-1 menu__list-item"><a class="menu__link" href="/labtutorial/Software/planemo">Planemo</a></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-1 menu__list-item"><a class="menu__link" href="/labtutorial/Software/cac">Cornell&#x27;s CAC server setup</a></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-1 menu__list-item"><a class="menu__link" href="/labtutorial/Software/frontend-dev">Frontend Deployment</a></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-1 menu__list-item"><a class="menu__link" href="/labtutorial/Software/backend-deployment">Backend Deployment</a></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-1 menu__list-item"><a class="menu__link" href="/labtutorial/Software/pegr">PEGR</a></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-1 menu__list-item"><a class="menu__link" href="/labtutorial/Software/labwebsite">Lab Website @Cornell</a></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-1 menu__list-item"><a class="menu__link" href="/labtutorial/Software/publication-checklist">Publication Checklist</a></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-1 menu__list-item"><a class="menu__link" href="/labtutorial/Software/faq">FAQ</a></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-1 menu__list-item"><a class="menu__link" href="/labtutorial/Software/condatutorial">Setting up conda for Pugh Lab</a></li></ul></nav></div></div></aside><main class="docMainContainer_gTbr"><div class="container padding-top--md padding-bottom--lg"><div class="row"><div class="col docItemCol_VOVn"><div class="docItemContainer_Djhp"><article><nav class="theme-doc-breadcrumbs breadcrumbsContainer_Z_bl" aria-label="Breadcrumbs"><ul class="breadcrumbs" itemscope="" itemtype="https://schema.org/BreadcrumbList"><li class="breadcrumbs__item"><a aria-label="Home page" class="breadcrumbs__link" href="/labtutorial/"><svg viewBox="0 0 24 24" class="breadcrumbHomeIcon_YNFT"><path d="M10 19v-5h4v5c0 .55.45 1 1 1h3c.55 0 1-.45 1-1v-7h1.7c.46 0 .68-.57.33-.87L12.67 3.6c-.38-.34-.96-.34-1.34 0l-8.36 7.53c-.34.3-.13.87.33.87H5v7c0 .55.45 1 1 1h3c.55 0 1-.45 1-1z" fill="currentColor"></path></svg></a></li><li itemscope="" itemprop="itemListElement" itemtype="https://schema.org/ListItem" class="breadcrumbs__item breadcrumbs__item--active"><span class="breadcrumbs__link" itemprop="name">Galaxy</span><meta itemprop="position" content="1"></li></ul></nav><div class="tocCollapsible_ETCw theme-doc-toc-mobile tocMobile_ITEo"><button type="button" class="clean-btn tocCollapsibleButton_TO0P">On this page</button></div><div class="theme-doc-markdown markdown"><header><h1>Galaxy</h1></header><h2 class="anchor anchorWithStickyNavbar_LWe7" id="dependencies">Dependencies<a href="#dependencies" class="hash-link" aria-label="Direct link to Dependencies" title="Direct link to Dependencies">​</a></h2><ul><li>Assuming you have installed the required <a href="http://127.0.0.1:8000/#development-tools" target="_blank" rel="noopener noreferrer">development tools for Mac OS</a></li><li>Install <a href="http://samtools.sourceforge.net/-Install" target="_blank" rel="noopener noreferrer">samtools</a>.<ul><li><code>brew install samtools</code></li></ul></li><li>Install <a href="https://yarnpkg.com/en/docs/install" target="_blank" rel="noopener noreferrer">yarn</a>.<ul><li><code>brew install yarn</code></li></ul></li></ul><hr><h2 class="anchor anchorWithStickyNavbar_LWe7" id="installing-galaxy">Installing Galaxy<a href="#installing-galaxy" class="hash-link" aria-label="Direct link to Installing Galaxy" title="Direct link to Installing Galaxy">​</a></h2><p><em>It is recommended that you clone galaxy to your <code>~/Desktop</code>, so that GALAXY&#x27;s internal paths are shorter and avoid conda throwing any errors that your path exceeds the maximum characters for tool install paths</em></p><p><em>here we are going to work with <code>17.09</code> release of GALAXY.</em></p><ul><li>Open a <code>terminal</code> and clone GALAXY to your <code>~/Desktop</code>.</li></ul><div class="codeBlockContainer_Ckt0 theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_biex"><pre tabindex="0" class="prism-code language-text codeBlock_bY9V thin-scrollbar"><code class="codeBlockLines_e6Vv"><span class="token-line" style="color:#393A34"><span class="token plain">cd ~/Desktop</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">git clone -b release_17.09 https://github.com/galaxyproject/galaxy.git</span><br></span></code></pre><div class="buttonGroup__atx"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_eSgA" aria-hidden="true"><svg class="copyButtonIcon_y97N" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_LjdS" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><ul><li>Start GALAXY</li></ul><div class="codeBlockContainer_Ckt0 theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_biex"><pre tabindex="0" class="prism-code language-text codeBlock_bY9V thin-scrollbar"><code class="codeBlockLines_e6Vv"><span class="token-line" style="color:#393A34"><span class="token plain">cd galaxy</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">sh run.sh</span><br></span></code></pre><div class="buttonGroup__atx"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_eSgA" aria-hidden="true"><svg class="copyButtonIcon_y97N" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_LjdS" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p><em>Note that when you are starting galaxy for the first time, it takes some time to install required internal dependencies and setup its internal database. Once its gone through it, you should be able to see the start up page.</em></p><ul><li>Congratulations you have successfully installed GALAXY on your local machine.</li></ul><table><thead><tr><th>Useful resources &amp; tutorials</th></tr></thead><tbody><tr><td><a href="https://galaxyproject.org/admin/get-galaxy/" target="_blank" rel="noopener noreferrer">Get galaxy</a></td></tr><tr><td><a href="https://galaxyproject.org/learn/" target="_blank" rel="noopener noreferrer">Learn galaxy</a></td></tr><tr><td><a href="https://github.com/galaxyproject/dagobah-training" target="_blank" rel="noopener noreferrer">Dagobah training</a></td></tr></tbody></table><hr><h2 class="anchor anchorWithStickyNavbar_LWe7" id="configuring-galaxy">Configuring Galaxy<a href="#configuring-galaxy" class="hash-link" aria-label="Direct link to Configuring Galaxy" title="Direct link to Configuring Galaxy">​</a></h2><p><strong>Setting up an Admin</strong></p><p><code>galaxy.ini</code> file contains most of the configurations to your local galaxy. It has some predefined defaults which you might want to re-think before changing anything.</p><p><em>We will be using <code>cegr@psu.edu</code> as the default admin, so that it can be used to integrate with PEGR in the PEGR tutorial, where we use our local galaxy to send out information to PEGR</em></p><ul><li>Make sure your Galaxy is not online, before making any changes to configuration.</li><li>Open your galaxy folder in <code>Finder</code> on your mac. you should see a <code>config/</code> folder.</li><li>Inside the <code>config/</code>, you will find a <code>galaxy.ini.sample</code> file.</li><li>Copy the <code>galaxy.ini.sample</code> within the same folder and rename the copied file to <code>galaxy.ini</code></li><li>Open the <code>galaxy.ini</code> in a text editor of your choice.</li><li>Search for the line that starts with <code>admin_users</code> and add <code>cegr@psu.edu</code></li></ul><p><img loading="lazy" alt="Galaxy Admin config Image" src="/labtutorial/assets/images/localgalaxy_23-8d027dbc03c575700de71c816c234a2e.png" width="686" height="145" class="img_ev3q"></p><p><em>Above image shows the change. make sure you have edited <code>~/Desktop/galaxy/config/galaxy.ini</code></em></p><ul><li>Start your Galaxy now, <code>sh run.sh</code></li><li>Click on <code>User</code> menu and then click <code>register</code></li><li>You should see a registration page as below</li></ul><p><img loading="lazy" alt="Galaxy register page" src="/labtutorial/assets/images/register-d989346c4694e1f1bfbb39b293707db9.png" width="2450" height="894" class="img_ev3q"></p><ul><li>Enter <code>cegr@psu.edu</code> as the email address and choose a <code>password</code>, <code>public name</code> of your choice.</li><li>Once you login, you should now have the <code>Admin</code> menu show up on the menu bar on the top.</li></ul><p><img loading="lazy" alt="Galaxy Admin option" src="/labtutorial/assets/images/adminmenu-d8861b7fa6675805e5ec5dae7c858a0b.png" width="1826" height="570" class="img_ev3q"></p><ul><li>When you click on the <code>Admin</code> tab, you should see the below page.</li></ul><p><img loading="lazy" alt="Galaxy Admin page" src="/labtutorial/assets/images/adminpage-d44a426881d92d84cfeac6966c59bc3f.png" width="3256" height="1966" class="img_ev3q"></p><hr><h2 class="anchor anchorWithStickyNavbar_LWe7" id="adding-custom-genomes">Adding Custom Genomes<a href="#adding-custom-genomes" class="hash-link" aria-label="Direct link to Adding Custom Genomes" title="Direct link to Adding Custom Genomes">​</a></h2><p><em><code>sacCer3_cegr</code> is the customized yeast genome we are using within the lab. Though the differences are few (to the best of my knowledge), but are very important to keep in mind while performing general data analysis. This genome deviates from the UCSC recommendations</em></p><p><em><code>sacCer3_cegr</code> contains <code>2-micron</code> regions &amp; chromosome naming is using <code>decimal number</code> instead of <code>roman numerals.</code> This causes some disadvantages to use tools like bedGraphToBigWig, etc.</em></p><ul><li>Click on <code>Admin</code> tab, then under <code>Tools and Tool Shed</code> section, click on <code>Search Tool Shed</code></li></ul><p><img loading="lazy" alt="Galaxy ToolShed" src="/labtutorial/assets/images/galaxy_toolshed-f7ca9da08d19e39db8583cb49db93e6f.png" width="2364" height="1968" class="img_ev3q"></p><ul><li>Click on <code>Galaxy Main Tool Shed</code></li></ul><p><img loading="lazy" alt="Galaxy ToolShed Categories" src="/labtutorial/assets/images/toolshed_categories-de64ec645bbe0c3c89ba1a84c2aadcaf.png" width="3262" height="2174" class="img_ev3q"></p><ul><li>Search &amp; Install below tools. you can copy and paste below tool names into the search box.</li></ul><table><thead><tr><th>Tool Name</th></tr></thead><tbody><tr><td><code>data_manager_bwa_mem_index_builder</code></td></tr><tr><td><code>data_manager_fetch_genome_dbkeys_all_fasta</code></td></tr><tr><td><code>data_manager_sam_fasta_index_builder</code></td></tr><tr><td><code>data_manager_twobit_builder</code></td></tr></tbody></table><ul><li>Below images previews the steps for installing a tool.</li></ul><p><img loading="lazy" alt="Tool Install ex1" src="/labtutorial/assets/images/install_ex1-f2c4403d88d876bf43e12d372e4fe22a.png" width="2954" height="468" class="img_ev3q"></p><p><img loading="lazy" alt="Tool Install ex2" src="/labtutorial/assets/images/install_ex2-5d0970b3ae027fb082181587accf0ff6.png" width="2946" height="776" class="img_ev3q"></p><p><img loading="lazy" alt="Tool Install ex3" src="/labtutorial/assets/images/install_ex3-02eaff6578dd78196c263b7c45c2fc0b.png" width="2950" height="876" class="img_ev3q"></p><ul><li>Once you have installed all the above tools, You can verify the installation at <code>Admin &gt; Tools and Tool shed &gt; Manage installed tools</code></li></ul><p><img loading="lazy" alt="Manage installed tools" src="/labtutorial/assets/images/localgalaxy_1-3f9f62de1b1f7a56b02c2a1cf4b55009.png" width="3152" height="1980" class="img_ev3q"></p><ul><li>Download the <code>sacCer3_cegr</code> genome from <a target="_blank" href="/labtutorial/assets/files/sacCer3_cegr-e2ff44617d4194178b35524392828ce9.fa">here</a></li><li>Upload the file you downloaded above into galaxy, using the <code>upload button</code> located on the tools menu, as shown in the image below.</li></ul><p><img loading="lazy" alt="Upload button location" src="/labtutorial/assets/images/upload_button-33f726ae93846e38b294a39912ab3c27.png" width="2960" height="1186" class="img_ev3q"></p><ul><li>Once you have uploaded, you should see the file appear in the “history pane” on your right.</li><li>Go to <code>Admin</code> tab, under <code>Data</code> section, Click on <code>Local data</code> to get <code>Data manager</code></li></ul><p><img loading="lazy" alt="Data Manager" src="/labtutorial/assets/images/localgalaxy_2-18f9a909614f3780e089576423d61b14.png" width="3154" height="2338" class="img_ev3q"></p><ul><li>We need to run below tools one after the other in the exact order mentioned below:</li></ul><table><thead><tr><th>Tool</th><th>Order</th></tr></thead><tbody><tr><td>Create DBkey and reference geneome</td><td>#1</td></tr><tr><td>TwoBit</td><td>#2</td></tr><tr><td>BWA-MEM</td><td>#3</td></tr><tr><td>SAM Fasta</td><td>#4</td></tr></tbody></table><p><em>All these tools are available from the <code>Admin &gt; Data &gt; Local data</code> section. Below are images for step-by-step execution of above tools in the same order. use them to fill out any default information that is required and follow along</em></p><ul><li>You can specify sequence name to be <code>sacCer3_cegr</code> and leave everything as default in all the tools.</li></ul><p><img loading="lazy" alt="Step 1" src="/labtutorial/assets/images/localgalaxy_3-bbf61981f717f92dcbd85e2519e4dbd1.png" width="3156" height="1396" class="img_ev3q">
<img loading="lazy" alt="Step 2" src="/labtutorial/assets/images/localgalaxy_4-c8f26802f65c98aecba104a547714d5e.png" width="3156" height="1134" class="img_ev3q">
<img loading="lazy" alt="Step 3" src="/labtutorial/assets/images/localgalaxy_5-daadad161d50e9b9f10f8f334a53bb6c.png" width="3154" height="1108" class="img_ev3q">
<img loading="lazy" alt="Step 4" src="/labtutorial/assets/images/localgalaxy_6-9a2f8027f74186bab58aa0f9c045f190.png" width="3152" height="1142" class="img_ev3q">
<img loading="lazy" alt="Step 5" src="/labtutorial/assets/images/localgalaxy_7-3572fd49ed92cbf84af559b93d4f348b.png" width="3152" height="1132" class="img_ev3q"></p><ul><li>Once you have run all the tools, you need to check couple of things that populate in the internal database of galaxy. You can check that information from your <code>Admin</code> page.</li><li>Go to <code>Admin &gt; Data &gt; Local data</code> section, under <code>View Tool Data Table Entries</code>. Click on <code>__dbkeys__</code> you should see something like below.</li></ul><p><img loading="lazy" alt="viewtools data table entries" src="/labtutorial/assets/images/localgalaxy_11-bf3e833f5d8d8fb60bd43059309d2604.png" width="2418" height="2194" class="img_ev3q"></p><p><img loading="lazy" alt="Step 6" src="/labtutorial/assets/images/localgalaxy_8-966efac156742a84e4ee4f808d03ced7.png" width="3158" height="404" class="img_ev3q"></p><ul><li>Similarly check <code>all_fasta</code>,<code>twobit</code>, <code>bwa_mem_indexes</code>. All of them should contain an entry for <code>sacCer3_cegr</code>. Path could be different in your case.</li></ul><p><img loading="lazy" alt="Step 7" src="/labtutorial/assets/images/localgalaxy_9-67c66ae29d991d1a95a0700e94ac4a33.png" width="3156" height="414" class="img_ev3q">
<img loading="lazy" alt="Step 8" src="/labtutorial/assets/images/localgalaxy_10-db0910de5169e1b097226c403816223e.png" width="3156" height="410" class="img_ev3q">
<img loading="lazy" alt="Step 9" src="/labtutorial/assets/images/localgalaxy_12-9dd3d272617f8832540618ed7e0bb8af.png" width="3224" height="226" class="img_ev3q"></p><ul><li>If your able to see similar results as above images. <code>Congratulations!</code> you have successfully added a custom genome build into your galaxy.</li></ul><hr><h2 class="anchor anchorWithStickyNavbar_LWe7" id="importing-chip-exo-workflow">Importing ChIP-exo Workflow<a href="#importing-chip-exo-workflow" class="hash-link" aria-label="Direct link to Importing ChIP-exo Workflow" title="Direct link to Importing ChIP-exo Workflow">​</a></h2><p><em>We will install the core-sequencing workflow that the lab uses to analyze all the samples that are sequenced. This pipeline is run to create BAM files, peak calling using genetrack and MEME motif analysis.</em></p><ul><li>Download the workflow file <a target="_blank" href="/labtutorial/assets/files/Galaxy-Workflow-paired_002-9e90b97c3853be35a38ae21997639277.ga">here.</a></li><li>Once you have downloaded the workflow. You can <code>import</code> it into your local galaxy from the <code>Workflow</code> tab using the <code>upload or import workflow</code> button located beside the search bar. <em>(see image below).</em></li></ul><p><img loading="lazy" alt="Workflow upload" src="/labtutorial/assets/images/localgalaxy_16-aaa33e83e32f756feccfdd51f9641f5f.png" width="3154" height="358" class="img_ev3q"></p><p><img loading="lazy" alt="Workflow import menu" src="/labtutorial/assets/images/localgalaxy_17-efdb31a543cd8453894a9ddd8eb7b27c.png" width="3152" height="856" class="img_ev3q"></p><ul><li>Once you have selected the workflow file and clicked <code>import</code>. You might see some errors such as below. Nothing to worry, the error messages is caused by tools that are not yet installed &amp; are important for the workflow to run in your galaxy.</li></ul><p><img loading="lazy" alt="Workflow import errors" src="/labtutorial/assets/images/localgalaxy_18-7898e9c08ea54cf980c70d10222c4f9a.png" width="3152" height="1278" class="img_ev3q"></p><ul><li>Click on <code>edit</code> option under the workflow drop-down menu. To find out the missing tools.</li><li>You need to install each tool manually from toolshed. Go to <code>Admin &gt; Tools and Tool Shed &gt; Search toolShed</code> to search and install each tool that is missing.</li></ul><p><em>Few tools have their toolnames in the workflow that end with <code>output_statistics</code>. These are the tools that are not available on <code>Galaxy toolshed</code> and need to be side-loaded separately, which we will do in the next section. so for now, you can ignore installing these tools and their errors.</em></p><hr><h2 class="anchor anchorWithStickyNavbar_LWe7" id="integrating-cegr-output_statistics">Integrating CEGR <em><code>output_statistics</code></em><a href="#integrating-cegr-output_statistics" class="hash-link" aria-label="Direct link to integrating-cegr-output_statistics" title="Direct link to integrating-cegr-output_statistics">​</a></h2><p><em>This section is similar to adding custom tools into Galaxy, here is a  (<a href="https://galaxyproject.org/admin/tools/add-tool-tutorial/" target="_blank" rel="noopener noreferrer">tutorial</a>). <code>cegr-galaxy</code> repo contains other important scripts that are used to run the core-sequencing pipeline on production galaxy. The repo has a <code>README</code> file detailing the usage of each script.</em></p><ul><li><p>Clone the repository containing the CEGR tools from <a href="https://github.com/seqcode/cegr-galaxy" target="_blank" rel="noopener noreferrer">seqcode/cegr-galaxy</a></p><ul><li><code>git clone https://github.com/seqcode/cegr-galaxy.git</code></li></ul></li><li><p>Copy the entire <code>cegr_statistics</code> folder to this location <code>galaxy/tools/</code> within your local galaxy.</p></li><li><p>Open <code>galaxy/config/tool_conf.xml</code> in a text editor of your choice.</p></li></ul><p><em>If the above file doesn&#x27;t exist, there should be a file in the same config directory called <code>tool_conf.xml.main</code>, copy and rename the file to <code>tool_conf.xml</code></em></p><ul><li>Add below lines at the end of file, within the <code>&lt;/toolbox&gt;</code> tag.</li></ul><div class="codeBlockContainer_Ckt0 theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_biex"><pre tabindex="0" class="prism-code language-text codeBlock_bY9V thin-scrollbar"><code class="codeBlockLines_e6Vv"><span class="token-line" style="color:#393A34"><span class="token plain">&lt;section id=&quot;cegr_tools&quot; name=&quot;CEGR&quot;&gt;</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    &lt;tool file=&quot;cegr_statistics/bam_to_scidx_output_stats.xml&quot; /&gt;</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    &lt;tool file=&quot;cegr_statistics/bedtools_intersectbed_output_stats.xml&quot; /&gt;</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    &lt;tool file=&quot;cegr_statistics/bwa_mem_output_stats_single.xml&quot; /&gt;</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    &lt;tool file=&quot;cegr_statistics/cwpair2_output_stats.xml&quot; /&gt;</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    &lt;tool file=&quot;cegr_statistics/extract_genomic_dna_output_stats.xml&quot; /&gt;</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    &lt;tool file=&quot;cegr_statistics/extract_genomic_dna_output_stats2.xml&quot; /&gt;</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    &lt;tool file=&quot;cegr_statistics/extract_genomic_dna_output_stats3.xml&quot; /&gt;</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    &lt;tool file=&quot;cegr_statistics/fasta_nucleotide_color_plot_output_stats.xml&quot; /&gt;</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    &lt;tool file=&quot;cegr_statistics/fastqc_output_stats.xml&quot; /&gt;</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    &lt;tool file=&quot;cegr_statistics/fastqc_output_stats2.xml&quot; /&gt;</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    &lt;tool file=&quot;cegr_statistics/genetrack_output_stats.xml&quot; /&gt;</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    &lt;tool file=&quot;cegr_statistics/input_dataset_r1_output_stats.xml&quot; /&gt;</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    &lt;tool file=&quot;cegr_statistics/input_dataset_r2_output_stats.xml&quot; /&gt;</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    &lt;tool file=&quot;cegr_statistics/mark_duplicates_bam_output_stats.xml&quot; /&gt;</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    &lt;tool file=&quot;cegr_statistics/meme_fimo_output_stats.xml&quot; /&gt;</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    &lt;tool file=&quot;cegr_statistics/meme_meme_output_stats.xml&quot; /&gt;</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    &lt;tool file=&quot;cegr_statistics/pe_histogram_output_stats.xml&quot; /&gt;</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    &lt;tool file=&quot;cegr_statistics/repeatmasker_wrapper_output_stats.xml&quot; /&gt;</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    &lt;tool file=&quot;cegr_statistics/repeatmasker_wrapper_output_stats2.xml&quot; /&gt;</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    &lt;tool file=&quot;cegr_statistics/samtool_filter2_output_stats.xml&quot; /&gt;</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    &lt;tool file=&quot;cegr_statistics/tag_pileup_frequency_output_stats.xml&quot; /&gt;</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  &lt;/section&gt;</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain" style="display:inline-block"></span><br></span></code></pre><div class="buttonGroup__atx"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_eSgA" aria-hidden="true"><svg class="copyButtonIcon_y97N" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_LjdS" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p><em>The above lines informs GALAXY, where it can find the tools and corresponding toolwrappers</em></p><ul><li>The file <code>galaxy/config/tool_config.xml</code> should look something like below:</li></ul><p><img loading="lazy" alt="Adding CEGR tools" src="/labtutorial/assets/images/addingcegrtools-693fb154f0a4c6e220c48def7d12b0f8.png" width="725" height="951" class="img_ev3q"></p><ul><li>Save the file and restart Galaxy. You should now see these tools appear under <code>Tools</code> menu within galaxy&#x27;s <code>Analyze Data</code> tab similar to below.</li></ul><p><img loading="lazy" alt="CEGR tools view" src="/labtutorial/assets/images/cegr_tools-391c6ad6f5bc30c360d9e4c5a3ee6091.png" width="2950" height="2212" class="img_ev3q"></p><ul><li><code>Congratulations!</code> you have successfully installed <code>output_statistics</code> tools into your galaxy.</li></ul><hr><h2 class="anchor anchorWithStickyNavbar_LWe7" id="connecting-galaxy-to-pegr">Connecting Galaxy to PEGR<a href="#connecting-galaxy-to-pegr" class="hash-link" aria-label="Direct link to Connecting Galaxy to PEGR" title="Direct link to Connecting Galaxy to PEGR">​</a></h2><p><em>Galaxy and PEGR communicate with each other using API keys. If you have not setup a local developmental PEGR. Set it up using these <a href="/labtutorial/Software/pegr">instructions</a> &amp; come back to this section</em></p><p><em>In the above section we installed <code>output_statistics</code> tools. These are the tools that send back information to <code>PEGR</code>. We did not configure the tools in the above section, which we will do in this section.</em></p><p><strong>Changing Galaxy&#x27;s port</strong></p><ul><li>First, we will configure <code>Galaxy</code> to run on a different <code>port</code> so that it does not conflict with <code>PEGR</code>&#x27;s default port.</li><li>Open <code>galaxy/config/galaxy.ini</code> in a text editor and search for <code>port</code> setting. Change the port to <code>8090</code>. Below image shows the change in <code>galaxy.ini</code>.</li></ul><p><img loading="lazy" alt="galaxy port change" src="/labtutorial/assets/images/galaxy_port_change-f843f82f91a65c96426847e23566d4ce.png" width="767" height="707" class="img_ev3q"></p><p><strong>Generating Galaxy API key</strong></p><ul><li>Click on <code>Admin</code> tab, under <code>User Management</code> section, click on <code>API keys</code>.</li><li>Click on <code>Generate</code> button to create an API key. Don&#x27;t regenerate a new key, if you already have one at this point.</li></ul><p><strong>CEGR <code>output_statistics</code> configuration</strong></p><ul><li>Open <code>galaxy/tools/stats_config.ini.sample</code> and add below information</li></ul><div class="language-ini codeBlockContainer_Ckt0 theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_biex"><pre tabindex="0" class="prism-code language-ini codeBlock_bY9V thin-scrollbar"><code class="codeBlockLines_e6Vv"><span class="token-line" style="color:#393A34"><span class="token plain" style="display:inline-block"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"># Configuration file for the CEGR Galaxy ChIP-exo statistics tools.</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain" style="display:inline-block"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">[defaults]</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain" style="display:inline-block"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"># This section contains default settings for command line parameters that</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"># can be overridden when they are passed to executed scripts.</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain" style="display:inline-block"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">PEGR_API_KEY =  &lt;REPLACE THIS WITH PEGR API KEY&gt;</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">PEGR_URL = http://localhost:8080/pegr/api/stats</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain" style="display:inline-block"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">GALAXY_API_KEY = &lt;REPLACE THIS WITH YOUR GALAXY ADMIN USER&#x27;s API KEY&gt;</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">GALAXY_BASE_URL = http://localhost:8090</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain" style="display:inline-block"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain" style="display:inline-block"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">[tool_categories]</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain" style="display:inline-block"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">input_dataset_r1 = output_fastqRead1</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">input_dataset_r2 = output_fastqRead2</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">toolshed.g2.bx.psu.edu/repos/iuc/bam_to_scidx/bam_to_scidx/1.0.1 = output_bamToScidx</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_intersectbed/2.27.0.0 = output_bedtoolsIntersect</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">toolshed.g2.bx.psu.edu/repos/devteam/bwa/bwa_mem/0.7.17.1 = output_bwaMem</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">toolshed.g2.bx.psu.edu/repos/iuc/cwpair2/cwpair2/1.1.0 = output_cwpair2</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">toolshed.g2.bx.psu.edu/repos/iuc/genetrack/genetrack/1.0.1 = output_genetrack</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">toolshed.g2.bx.psu.edu/repos/iuc/extract_genomic_dna/Extract genomic DNA 1/3.0.3 = output_extractGenomicDNA</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.70 = output_fastqc</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">toolshed.g2.bx.psu.edu/repos/iuc/pe_histogram/pe_histogram/1.0.1 = output_peHistogram</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">toolshed.g2.bx.psu.edu/repos/bgruening/repeat_masker/repeatmasker_wrapper/0.1.2 =output_repeatMasker</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">toolshed.g2.bx.psu.edu/repos/iuc/meme_meme/meme_meme/4.11.2.0 = output_meme</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">toolshed.g2.bx.psu.edu/repos/iuc/fasta_nucleotide_color_plot/fasta_nucleotide_color_plot/1.0.1 = output_fourColorPlot</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">toolshed.g2.bx.psu.edu/repos/jjohnson/samtools_filter/samtools_filter/1.1.1 = output_samtoolFilter</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">toolshed.g2.bx.psu.edu/repos/devteam/picard/picard_MarkDuplicates/2.7.1.1 = output_markDuplicates</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">toolshed.g2.bx.psu.edu/repos/iuc/meme_fimo/meme_fimo/4.11.2.0 = output_fimo</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">toolshed.g2.bx.psu.edu/repos/iuc/tag_pileup_frequency/tag_pileup_frequency/1.0.1 = output_tagPileup</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain" style="display:inline-block"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain" style="display:inline-block"></span><br></span></code></pre><div class="buttonGroup__atx"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_eSgA" aria-hidden="true"><svg class="copyButtonIcon_y97N" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_LjdS" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><ul><li>Below is an example configuration, after you add your API Keys.</li></ul><p><img loading="lazy" alt="Example config" src="/labtutorial/assets/images/sample_stats_config-4307cdf5c042648d38afbdb38d91f4cc.png" width="883" height="739" class="img_ev3q"></p><ul><li>Rename <code>stats_config.ini.sample</code> to <code>stats_config.ini</code></li></ul><p><strong>Install bioblend</strong></p><p><em>bioblend is a python library to interact with Galaxy, using APIs</em></p><ul><li>Installation instructions <a href="https://bioblend.readthedocs.io/en/latest/#installation" target="_blank" rel="noopener noreferrer">here</a></li><li>If you have already installed <a href="https://www.anaconda.com/" target="_blank" rel="noopener noreferrer"><code>Anaconda</code></a> &amp; <a href="https://pypi.org/project/pip/" target="_blank" rel="noopener noreferrer"><code>pip</code></a><ul><li><code>pip install bioblend</code></li></ul></li><li>Using bioblend <a href="https://bioblend.readthedocs.io/en/latest/api_docs/galaxy/docs.html" target="_blank" rel="noopener noreferrer">docs</a></li></ul><p><strong>Adding core-sequencing workflowId into your local development PEGRdb</strong></p><p><em>This step is important, as it lets PEGR know which workflow stats to accept and update on PEGR web frontend</em></p><ul><li>Download <code>getWorkflowid.py</code> from <a target="_blank" href="/labtutorial/assets/files/getWorkflowid-f94fa3d8a55ac011423a88be16cecd04.py">here</a></li><li>Open the script and replace the <code>url</code> and <code>key</code> with your local galaxy url and API key respectively. After adding your API-key, looks like the below image:</li></ul><p><img loading="lazy" alt="Get workflowid" src="/labtutorial/assets/images/getworkflowid-ff919e1386ed012e08513ca9b08d74c4.png" width="795" height="349" class="img_ev3q"></p><ul><li>Start your galaxy</li><li>Run the script to retrieve the workflowId. Below is an image showing expected output</li></ul><p><img loading="lazy" alt="expected output for get workflowid" src="/labtutorial/assets/images/localgalaxy_27-05b3dff9b25bc784ee821687e509cbe7.png" width="1028" height="518" class="img_ev3q"></p><ul><li>Your workflow id is <code>ebfb8f50c6abde6d</code> in the above example.</li><li>Start your <code>MySQL</code> server and log into it using the username and password that you used while setting up <code>PEGR</code> or you can login as root  <code>mysql -u root -p</code></li></ul><div class="codeBlockContainer_Ckt0 theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_biex"><pre tabindex="0" class="prism-code language-text codeBlock_bY9V thin-scrollbar"><code class="codeBlockLines_e6Vv"><span class="token-line" style="color:#393A34"><span class="token plain">use pegr;</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">describe pipeline;</span><br></span></code></pre><div class="buttonGroup__atx"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_eSgA" aria-hidden="true"><svg class="copyButtonIcon_y97N" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_LjdS" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p><img loading="lazy" alt="example query" src="/labtutorial/assets/images/localgalaxy_28-5ab81ced59a7fab530650c8266d94b60.png" width="1232" height="444" class="img_ev3q"></p><ul><li>Edit below query with a <code>pipeline-name</code> of your choice and add the <code>workflowid</code> you retrieved from above script and execute it. Below is an example query that you need</li></ul><div class="codeBlockContainer_Ckt0 theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_biex"><pre tabindex="0" class="prism-code language-text codeBlock_bY9V thin-scrollbar"><code class="codeBlockLines_e6Vv"><span class="token-line" style="color:#393A34"><span class="token plain">insert into pipeline( version,name,note,pipeline_version,steps,workflow_id)</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">values( 1,</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain" style="display:inline-block"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">&#x27;&lt;pipeline-name&gt;&#x27;,</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain" style="display:inline-block"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">&#x27;Locally installed galaxy sending JSON dictionaires to locally installed pegr&#x27;,&#x27;001&#x27;, &#x27;[[&quot;input_dataset_r1_output_stats&quot;,&quot;fastqRead1&quot;],[&quot;input_dataset_r2_output_stats&quot;,&quot;fastqRead2&quot;],[&quot;fastqc_output_stats&quot;,&quot;fastqc&quot;],[&quot;fastqc_output_stats2&quot;,&quot;fastqc&quot;],[&quot;mark_duplicates_bam_output_stats&quot;,&quot;markDuplicates&quot;],[&quot;samtool_filter2_output_stats&quot;,&quot;samtoolFilter&quot;],[&quot;pe_histogram_output_stats&quot;,&quot;peHistogram&quot;],[&quot;bam_to_scidx_output_stats&quot;,&quot;bamToScidx&quot;],[&quot;genetrack_output_stats&quot;,&quot;genetrack&quot;],[&quot;bedtools_intersectbed_output_stats&quot;,&quot;bedtoolsIntersect&quot;],[&quot;cwpair2_output_stats&quot;,&quot;cwpair2&quot;],[&quot;extract_genomic_dna_output_stats&quot;,&quot;extractGenomicDNA&quot;],[&quot;extract_genomic_dna_output_stats2&quot;,&quot;extractGenomicDNA&quot;],[&quot;repeatmasker_wrapper_output_stats&quot;,&quot;repeatMasker&quot;],[&quot;repeatmasker_wrapper_output_stats2&quot;,&quot;repeatMasker&quot;],[&quot;meme_meme_output_stats&quot;,&quot;meme&quot;],[&quot;meme_fimo_output_stats&quot;,&quot;fimo&quot;],[&quot;extract_genomic_dna_output_stats3&quot;,&quot;extractGenomicDNA&quot;],[&quot;fasta_nucleotide_color_plot_output_stats&quot;,&quot;fourColorPlot&quot;],[&quot;tag_pileup_frequency_output_stats&quot;,&quot;tagPileup&quot;]]&#x27; ,</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain" style="display:inline-block"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">&quot;&lt;workflowid&gt;&quot;);</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain" style="display:inline-block"></span><br></span></code></pre><div class="buttonGroup__atx"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_eSgA" aria-hidden="true"><svg class="copyButtonIcon_y97N" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_LjdS" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><ul><li>This how your query looks like after adding your pipeline-name and workflowid</li></ul><p><img loading="lazy" alt="query" src="/labtutorial/assets/images/localgalaxy_29-176027f8c328270e8cb042b3fb52845d.png" width="1936" height="372" class="img_ev3q"></p><ul><li>You are all set for executing the pipeline. <em>(if you have followed and set up the keys correctly, there should be no errors)</em></li></ul></div><footer class="theme-doc-footer docusaurus-mt-lg"><div class="theme-doc-footer-edit-meta-row row"><div class="col"></div><div class="col lastUpdated_vwxv"><span class="theme-last-updated">Last updated<!-- --> on <b><time datetime="2023-05-02T21:22:01.000Z">May 2, 2023</time></b> by <b>owlang</b></span></div></div></footer></article><nav class="pagination-nav docusaurus-mt-lg" aria-label="Docs pages navigation"><a class="pagination-nav__link pagination-nav__link--next" href="/labtutorial/Software/planemo"><div class="pagination-nav__sublabel">Next</div><div class="pagination-nav__label">Planemo</div></a></nav></div></div><div class="col col--3"><div class="tableOfContents_bqdL thin-scrollbar theme-doc-toc-desktop"><ul class="table-of-contents table-of-contents__left-border"><li><a href="#dependencies" class="table-of-contents__link toc-highlight">Dependencies</a></li><li><a href="#installing-galaxy" class="table-of-contents__link toc-highlight">Installing Galaxy</a></li><li><a href="#configuring-galaxy" class="table-of-contents__link toc-highlight">Configuring Galaxy</a></li><li><a href="#adding-custom-genomes" class="table-of-contents__link toc-highlight">Adding Custom Genomes</a></li><li><a href="#importing-chip-exo-workflow" class="table-of-contents__link toc-highlight">Importing ChIP-exo Workflow</a></li><li><a href="#integrating-cegr-output_statistics" class="table-of-contents__link toc-highlight">Integrating CEGR <em><code>output_statistics</code></em></a></li><li><a href="#connecting-galaxy-to-pegr" class="table-of-contents__link toc-highlight">Connecting Galaxy to PEGR</a></li></ul></div></div></div></div></main></div></div><footer class="footer footer--dark"><div class="container container-fluid"><div class="row footer__links"><div class="col footer__col"><div class="footer__title">Docs</div><ul class="footer__items clean-list"><li class="footer__item"><a class="footer__link-item" href="/labtutorial/index">Wiki Home</a></li></ul></div><div class="col footer__col"><div class="footer__title">Community</div><ul class="footer__items clean-list"><li class="footer__item"><a href="https://pughlab.mbg.cornell.edu" target="_blank" rel="noopener noreferrer" class="footer__link-item">Pugh Lab<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_nPIU"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></a></li><li class="footer__item"><a href="https://williamkmlai.github.io" target="_blank" rel="noopener noreferrer" class="footer__link-item">Lai Lab<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_nPIU"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></a></li></ul></div><div class="col footer__col"><div class="footer__title">Other Tools We Develop</div><ul class="footer__items clean-list"><li class="footer__item"><a href="http://pughlab.mbg.cornell.edu/scriptmanager" target="_blank" rel="noopener noreferrer" class="footer__link-item">ScriptManager<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_nPIU"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></a></li><li class="footer__item"><a href="https://github.com/seqcode/pegr" target="_blank" rel="noopener noreferrer" class="footer__link-item">PEGR<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_nPIU"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></a></li><li class="footer__item"><a href="http://pughlab.mbg.cornell.edu/stencil" target="_blank" rel="noopener noreferrer" class="footer__link-item">STENCIL<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_nPIU"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></a></li><li class="footer__item"><a href="http://pughlab.mbg.cornell.edu/GenoPipe" target="_blank" rel="noopener noreferrer" class="footer__link-item">GenoPipe<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_nPIU"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></a></li></ul></div></div><div class="footer__bottom text--center"><div class="footer__copyright">Copyright © 2023 labtutorial. Built with Docusaurus.</div></div></div></footer></div>
<script src="/labtutorial/assets/js/runtime~main.0d187c5d.js"></script>
<script src="/labtutorial/assets/js/main.3364e39b.js"></script>
</body>
</html>