"use strict";(self.webpackChunklabtutorial=self.webpackChunklabtutorial||[]).push([[838],{5680:(e,t,a)=>{a.d(t,{xA:()=>u,yg:()=>g});var i=a(6540);function o(e,t,a){return t in e?Object.defineProperty(e,t,{value:a,enumerable:!0,configurable:!0,writable:!0}):e[t]=a,e}function s(e,t){var a=Object.keys(e);if(Object.getOwnPropertySymbols){var i=Object.getOwnPropertySymbols(e);t&&(i=i.filter((function(t){return Object.getOwnPropertyDescriptor(e,t).enumerable}))),a.push.apply(a,i)}return a}function n(e){for(var t=1;t<arguments.length;t++){var a=null!=arguments[t]?arguments[t]:{};t%2?s(Object(a),!0).forEach((function(t){o(e,t,a[t])})):Object.getOwnPropertyDescriptors?Object.defineProperties(e,Object.getOwnPropertyDescriptors(a)):s(Object(a)).forEach((function(t){Object.defineProperty(e,t,Object.getOwnPropertyDescriptor(a,t))}))}return e}function r(e,t){if(null==e)return{};var a,i,o=function(e,t){if(null==e)return{};var a,i,o={},s=Object.keys(e);for(i=0;i<s.length;i++)a=s[i],t.indexOf(a)>=0||(o[a]=e[a]);return o}(e,t);if(Object.getOwnPropertySymbols){var s=Object.getOwnPropertySymbols(e);for(i=0;i<s.length;i++)a=s[i],t.indexOf(a)>=0||Object.prototype.propertyIsEnumerable.call(e,a)&&(o[a]=e[a])}return o}var l=i.createContext({}),p=function(e){var t=i.useContext(l),a=t;return e&&(a="function"==typeof e?e(t):n(n({},t),e)),a},u=function(e){var t=p(e.components);return i.createElement(l.Provider,{value:t},e.children)},c="mdxType",d={inlineCode:"code",wrapper:function(e){var t=e.children;return i.createElement(i.Fragment,{},t)}},m=i.forwardRef((function(e,t){var a=e.components,o=e.mdxType,s=e.originalType,l=e.parentName,u=r(e,["components","mdxType","originalType","parentName"]),c=p(a),m=o,g=c["".concat(l,".").concat(m)]||c[m]||d[m]||s;return a?i.createElement(g,n(n({ref:t},u),{},{components:a})):i.createElement(g,n({ref:t},u))}));function g(e,t){var a=arguments,o=t&&t.mdxType;if("string"==typeof e||o){var s=a.length,n=new Array(s);n[0]=m;var r={};for(var l in t)hasOwnProperty.call(t,l)&&(r[l]=t[l]);r.originalType=e,r[c]="string"==typeof e?e:o,n[1]=r;for(var p=2;p<s;p++)n[p]=a[p];return i.createElement.apply(null,n)}return i.createElement.apply(null,a)}m.displayName="MDXCreateElement"},6346:(e,t,a)=>{a.r(t),a.d(t,{assets:()=>l,contentTitle:()=>n,default:()=>d,frontMatter:()=>s,metadata:()=>r,toc:()=>p});var i=a(8168),o=(a(6540),a(5680));const s={title:"GEO Submission",sidebar_label:"GEO Submission",sidebar_position:4},n=void 0,r={unversionedId:"General/PublicationChecklist/geo-submission",id:"General/PublicationChecklist/geo-submission",title:"GEO Submission",description:"All data not subject to privacy restrictions (e.g. patient-protected samples) should be made publicly available at one of the national genomic data repositories. We submit through Gene Expression Omnibus (GEO).",source:"@site/docs/General/PublicationChecklist/geo-submission.md",sourceDirName:"General/PublicationChecklist",slug:"/General/PublicationChecklist/geo-submission",permalink:"/labtutorial/General/PublicationChecklist/geo-submission",draft:!1,tags:[],version:"current",lastUpdatedBy:"Olivia Wen-Mei Lang",lastUpdatedAt:1715885221,formattedLastUpdatedAt:"May 16, 2024",sidebarPosition:4,frontMatter:{title:"GEO Submission",sidebar_label:"GEO Submission",sidebar_position:4},sidebar:"generalBenchSidebar",previous:{title:"Github repo",permalink:"/labtutorial/General/PublicationChecklist/github"},next:{title:"Zenodo Submission",permalink:"/labtutorial/General/PublicationChecklist/zenodo"}},l={},p=[{value:"Account Set-up",id:"account-set-up",level:2},{value:"Get the latest instructions",id:"get-the-latest-instructions",level:2},{value:"Get your list of PEGR sample IDs to submit",id:"get-your-list-of-pegr-sample-ids-to-submit",level:2},{value:"From Project ID",id:"from-project-id",level:3},{value:"From Sample List",id:"from-sample-list",level:3},{value:"Retrieve data from PEGR",id:"retrieve-data-from-pegr",level:2},{value:"(Step 2) Gather raw files (FASTQ)",id:"step-2-gather-raw-files-fastq",level:3},{value:"(Step 3) Gather processed files",id:"step-3-gather-processed-files",level:3},{value:"(Step 4) Gather metadata",id:"step-4-gather-metadata",level:3},{value:"(Step 5) Get MD5 Checksums for all raw and processed files",id:"step-5-get-md5-checksums-for-all-raw-and-processed-files",level:2},{value:"(Step 6) Organize files according to GEO instructions",id:"step-6-organize-files-according-to-geo-instructions",level:2},{value:"(Step 7) Upload files to GEO servers",id:"step-7-upload-files-to-geo-servers",level:2},{value:"1. Transfer files",id:"1-transfer-files",level:3},{value:"2. Submit metadata",id:"2-submit-metadata",level:3},{value:"Follow up on the submission",id:"follow-up-on-the-submission",level:2},{value:"After manuscript is accepted...",id:"after-manuscript-is-accepted",level:2}],u={toc:p},c="wrapper";function d(e){let{components:t,...s}=e;return(0,o.yg)(c,(0,i.A)({},u,s,{components:t,mdxType:"MDXLayout"}),(0,o.yg)("p",null,"All data not subject to privacy restrictions (e.g. patient-protected samples) should be made publicly available at one of the national genomic data repositories. We submit through ",(0,o.yg)("a",{parentName:"p",href:"https://www.ncbi.nlm.nih.gov/geo/"},"Gene Expression Omnibus (GEO)"),"."),(0,o.yg)("admonition",{type:"danger"},(0,o.yg)("p",{parentName:"admonition"},"DO NOT PROCEED unless you are ready to submit your manuscript. It is a pain in the butt to edit a submission later so it is better to finalize your datasets beforehand.")),(0,o.yg)("p",null,"When the data used for your manuscript is somewhat finalized, follow the instructions here to submit your data. Allow for a couple days time for curators at NCBI to process samples but they usually have a quick turnaround time."),(0,o.yg)("h2",{id:"account-set-up"},"Account Set-up"),(0,o.yg)("p",null,"You will need a MyNCBI account if you don't already have one. It should be linked to your ORCiD or Institution account. Make sure Frank's information is included as your PI with all of the usual address and contact information."),(0,o.yg)("h2",{id:"get-the-latest-instructions"},"Get the latest instructions"),(0,o.yg)("p",null,"Follow the ",(0,o.yg)("a",{parentName:"p",href:"https://www.ncbi.nlm.nih.gov/geo/info/submission.html"},"GEO submission instructions"),' for submitting the UNPUBLISHED datasets ("Novel Set") used in the manuscript. Almost all data we produce are ',(0,o.yg)("strong",{parentName:"p"},'"high-throughput sequencing data"')," if they are coming from PEGR."),(0,o.yg)("a",{href:"https://www.ncbi.nlm.nih.gov/geo/info/seq.html"},(0,o.yg)("div",{class:"tutorial-img-flow-container"},(0,o.yg)("img",{src:a(8514).A,style:{width:"100%"}}))),(0,o.yg)("admonition",{type:"note"},(0,o.yg)("p",{parentName:"admonition"},"The instructions from GEO always supercede the documentation on this page. Please talk to a bioinformatician if you run into inconsistencies between the GEO instructions and this page.")),(0,o.yg)("h2",{id:"get-your-list-of-pegr-sample-ids-to-submit"},"Get your list of PEGR sample IDs to submit"),(0,o.yg)("h3",{id:"from-project-id"},"From Project ID"),(0,o.yg)("p",null,(0,o.yg)("a",{parentName:"p",href:"https://vesta.cac.cornell.edu/pegr/project/show/1331"},"1331")," is the ID for ",(0,o.yg)("inlineCode",{parentName:"p"},"Krietenstein 2016 - Cell (Novel Set)"),". Check the end of the project URL if you're not sure you have the right project ID (vesta.cac.cornell.edu/pegr/project/show/",(0,o.yg)("strong",{parentName:"p"},"1331"),")"),(0,o.yg)("h3",{id:"from-sample-list"},"From Sample List"),(0,o.yg)("p",null,'Or you can get a list of sample IDs by scrolling down to the "Experiments" section and clicking "Export Samples to CSV"'),(0,o.yg)("p",null,(0,o.yg)("img",{alt:"PEGR_ExportSamplesFromProject",src:a(1454).A,width:"708",height:"290"})),(0,o.yg)("admonition",{type:"caution"},(0,o.yg)("p",{parentName:"admonition"},"Just use the ",(0,o.yg)("a",{parentName:"p",href:"./pegr-projects#novel-set"},'"Novel set"')," samples. Previously published samples (",(0,o.yg)("a",{parentName:"p",href:"./pegr-projects#reanalyzed-set"},'"Reanalyzed set"'),") that have been reanalyzed for this manuscript should NOT be included in the submission. These can be cited within the manuscript using their existing NCBI accession info.")),(0,o.yg)("h2",{id:"retrieve-data-from-pegr"},"Retrieve data from PEGR"),(0,o.yg)("p",null,"PEGR houses the metadata, raw data file links, and processed data file links that you need to create the GEO submission. We have built command line tools for facilitating the bulk retreival of this data."),(0,o.yg)("admonition",{type:"danger"},(0,o.yg)("p",{parentName:"admonition"},"Please do not manually fill in the GEO submission form or manually download files as that has historically been mistake-prone. Use the PEGR API (see links below).")),(0,o.yg)("h3",{id:"step-2-gather-raw-files-fastq"},"(Step 2) Gather raw files (FASTQ)"),(0,o.yg)("p",null,"Use ",(0,o.yg)("a",{parentName:"p",href:"https://github.com/CEGRcode/EGC_utility_scripts/blob/master/generate_FQ_file_from_PEGR.py"},"EGC_utility_scripts")," to pull the raw FASTQ sample files"),(0,o.yg)("h3",{id:"step-3-gather-processed-files"},"(Step 3) Gather processed files"),(0,o.yg)("p",null,'Samples may be "processed" differently depending on the data type. Use your judgement on what is appropriate for the purposes of the study. Consider referencing ',(0,o.yg)("a",{parentName:"p",href:"https://github.com/CEGRcode/2022-Mittal_SAGA/blob/main/03_Bulk_Processing/job/bulk_geo_tracks.pbs"},"this script")," for creating BigWig pileups of 5' Read1 coverage across the genome. Or you can pull the BigWig files from the core pipeline (",(0,o.yg)("inlineCode",{parentName:"p"},"paired_004"),"). If you are using ",(0,o.yg)("inlineCode",{parentName:"p"},"paired_004")," BigWigs, note the computational steps from the pipeline and fill in the GEO submission accordingly."),(0,o.yg)("h3",{id:"step-4-gather-metadata"},"(Step 4) Gather metadata"),(0,o.yg)("p",null,"Use ",(0,o.yg)("a",{parentName:"p",href:"https://github.com/CEGRcode/EGC_utility_scripts/blob/master/generate_Metadata_file_from_PEGR.py"},"EGC_utility_scripts")," for pulling metadata from PEGR as a starting point for building the spreadsheet. ",(0,o.yg)("strong",{parentName:"p"},"You may need to manually adjust the metadata output of this script to fit GEO specifications for the submission.")),(0,o.yg)("p",null,"See GEO for downloading the metadata template on the instructions page. Fill it in with the sample metadata and manuscript information."),(0,o.yg)("admonition",{type:"tip"},(0,o.yg)("p",{parentName:"admonition"},"Check Box for previous GEO submission spreadsheets.")),(0,o.yg)("h2",{id:"step-5-get-md5-checksums-for-all-raw-and-processed-files"},"(Step 5) Get MD5 Checksums for all raw and processed files"),(0,o.yg)("p",null,"You can use ",(0,o.yg)("a",{parentName:"p",href:"https://pughlab.mbg.cornell.edu/scriptmanager-docs/docs/Tools/file-utilities/md5checksum"},"ScriptManager")," to do this for all your FASTQ and processed (e.g. BigWig) files."),(0,o.yg)("h2",{id:"step-6-organize-files-according-to-geo-instructions"},"(Step 6) Organize files according to GEO instructions"),(0,o.yg)("p",null,"Follow the GEO instructions for organizing your data. Please separate data from different assays as appropriate. For example, separate RNA-seq from ChIP-exo or ATAC-seq data."),(0,o.yg)("h2",{id:"step-7-upload-files-to-geo-servers"},"(Step 7) Upload files to GEO servers"),(0,o.yg)("img",{src:a(6375).A,style:{width:"100%"}}),(0,o.yg)("h3",{id:"1-transfer-files"},"1. Transfer files"),(0,o.yg)("p",null,"Then upload the data to their servers. They give you a bunch of options for uploading these data files. If you aren't comfortable with FTP, try the FileZilla method (they have installation instructions if you don't already have it)."),(0,o.yg)("img",{src:a(269).A,style:{width:"100%"}}),(0,o.yg)("h3",{id:"2-submit-metadata"},"2. Submit metadata"),(0,o.yg)("p",null,"Select the folder name that you transfered and upload the excel file you prepared."),(0,o.yg)("admonition",{type:"caution"},(0,o.yg)("p",{parentName:"admonition"},"Regarding the date to make the data public, ",(0,o.yg)("strong",{parentName:"p"},"start with at least a year out"),". This can be changed later.")),(0,o.yg)("h2",{id:"follow-up-on-the-submission"},"Follow up on the submission"),(0,o.yg)("p",null,"Once approved, you should get an email from GEO with the newly created ",(0,o.yg)("inlineCode",{parentName:"p"},"GSEXXXXXX")," and ",(0,o.yg)("inlineCode",{parentName:"p"},"GSMXXXXXX")," accessions to include in the manuscript."),(0,o.yg)("ul",null,(0,o.yg)("li",{parentName:"ul"},"This typically means including the study (",(0,o.yg)("inlineCode",{parentName:"li"},"GSEXXXXXX"),") accession in the ",(0,o.yg)("strong",{parentName:"li"},"Methods")," or ",(0,o.yg)("strong",{parentName:"li"},"Data Access")," sections and the specific samples (",(0,o.yg)("inlineCode",{parentName:"li"},"GSMXXXXXX"),") in the ",(0,o.yg)("strong",{parentName:"li"},"extended methods/tables"),". The accessions of reanalyzed data from previously published samples should be similarly included.")),(0,o.yg)("p",null,"Also update with accession:"),(0,o.yg)("ul",null,(0,o.yg)("li",{parentName:"ul"},"Github repo"),(0,o.yg)("li",{parentName:"ul"},"PEGR projects (Novel, reanalyzed, and working)")),(0,o.yg)("h2",{id:"after-manuscript-is-accepted"},"After manuscript is accepted..."),(0,o.yg)("p",null,"Notify GEO so they can coordinate with the journal to release the data with the manuscript. We don't want the manuscript to be in print but not have the data available to readers!"))}d.isMDXComponent=!0},269:(e,t,a)=>{a.d(t,{A:()=>i});const i=a.p+"assets/images/GEO_FTPOptions-810307653ff0d7a72fa46d98c384078a.png"},8514:(e,t,a)=>{a.d(t,{A:()=>i});const i=a.p+"assets/images/GEO_SelectHTS-3fba2a58eb6d6477235aee4957217b18.png"},6375:(e,t,a)=>{a.d(t,{A:()=>i});const i=a.p+"assets/images/GEO_UploadYourSubmission-d445811ecdf238b78a4bef973241a564.png"},1454:(e,t,a)=>{a.d(t,{A:()=>i});const i=a.p+"assets/images/PEGR_ExportSamplesFromProject-080be9808d65a15ce815a61cd8c18882.png"}}]);