<!doctype html>
<html lang="en" dir="ltr" class="docs-wrapper docs-doc-page docs-version-current plugin-docs plugin-id-default docs-doc-id-General/PublicationChecklist/geo-submission">
<head>
<meta charset="UTF-8">
<meta name="generator" content="Docusaurus v2.4.0">
<title data-rh="true">GEO Submission | Lab Wiki</title><meta data-rh="true" name="viewport" content="width=device-width,initial-scale=1"><meta data-rh="true" name="twitter:card" content="summary_large_image"><meta data-rh="true" property="og:url" content="https://CEGRcode.github.io/labtutorial/General/PublicationChecklist/geo-submission"><meta data-rh="true" name="docusaurus_locale" content="en"><meta data-rh="true" name="docsearch:language" content="en"><meta data-rh="true" name="docusaurus_version" content="current"><meta data-rh="true" name="docusaurus_tag" content="docs-default-current"><meta data-rh="true" name="docsearch:version" content="current"><meta data-rh="true" name="docsearch:docusaurus_tag" content="docs-default-current"><meta data-rh="true" property="og:title" content="GEO Submission | Lab Wiki"><meta data-rh="true" name="description" content="All data not subject to privacy restrictions (e.g. patient-protected samples) should be made publicly available at one of the national genomic data repositories. We submit through Gene Expression Omnibus (GEO)."><meta data-rh="true" property="og:description" content="All data not subject to privacy restrictions (e.g. patient-protected samples) should be made publicly available at one of the national genomic data repositories. We submit through Gene Expression Omnibus (GEO)."><link data-rh="true" rel="icon" href="/labtutorial/img/favicon.png"><link data-rh="true" rel="canonical" href="https://CEGRcode.github.io/labtutorial/General/PublicationChecklist/geo-submission"><link data-rh="true" rel="alternate" href="https://CEGRcode.github.io/labtutorial/General/PublicationChecklist/geo-submission" hreflang="en"><link data-rh="true" rel="alternate" href="https://CEGRcode.github.io/labtutorial/General/PublicationChecklist/geo-submission" hreflang="x-default"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.13.24/dist/katex.min.css" integrity="sha384-odtC+0UGzzFL/6PNoE8rX/SPcQDXBJ+uRepguP4QkPCm2LBxH3FA3y+fKSiJ+AmM" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.3.0/css/all.min.css" integrity="sha512-SzlrxWUlpfuzQ+pcUCosxcglQRNAq/DZjVsC0lE40xsADsfeQoEypE+enwcOiGjk/bSuGGKHEyjSoQ1zVisanQ==" crossorigin="anonymous"><link rel="stylesheet" href="/labtutorial/assets/css/styles.e7b83035.css">
<link rel="preload" href="/labtutorial/assets/js/runtime~main.78b75138.js" as="script">
<link rel="preload" href="/labtutorial/assets/js/main.14ee8d51.js" as="script">
</head>
<body class="navigation-with-keyboard">
<script>!function(){function t(t){document.documentElement.setAttribute("data-theme",t)}var e=function(){var t=null;try{t=new URLSearchParams(window.location.search).get("docusaurus-theme")}catch(t){}return t}()||function(){var t=null;try{t=localStorage.getItem("theme")}catch(t){}return t}();t(null!==e?e:"light")}()</script><div id="__docusaurus">
<div role="region" aria-label="Skip to main content"><a class="skipToContent_fXgn" href="#docusaurus_skipToContent_fallback">Skip to main content</a></div><nav aria-label="Main" class="navbar navbar--fixed-top"><div class="navbar__inner"><div class="navbar__items"><button aria-label="Toggle navigation bar" aria-expanded="false" class="navbar__toggle clean-btn" type="button"><svg width="30" height="30" viewBox="0 0 30 30" aria-hidden="true"><path stroke="currentColor" stroke-linecap="round" stroke-miterlimit="10" stroke-width="2" d="M4 7h22M4 15h22M4 23h22"></path></svg></button><a href="http://pughlab.mbg.cornell.edu/" target="_blank" rel="noopener noreferrer" class="navbar__brand"><div class="navbar__logo"><img src="/labtutorial/img/logo.svg" alt="Lab logo" class="themedImage_ToTc themedImage--light_HNdA"><img src="/labtutorial/img/logo_dark.svg" alt="Lab logo" class="themedImage_ToTc themedImage--dark_i4oU"></div></a><a aria-current="page" class="navbar__item navbar__link navbar__link--active" href="/labtutorial/General/FigureMaking/">General</a><a class="navbar__item navbar__link" href="/labtutorial/Software/">Software</a><a class="navbar__item navbar__link" href="/labtutorial/WetBench/">Wet Bench</a></div><div class="navbar__items navbar__items--right"><a href="https://github.com/CEGRcode" target="_blank" rel="noopener noreferrer" class="navbar__item navbar__link header-github-link"></a><div class="toggle_vylO colorModeToggle_DEke"><button class="clean-btn toggleButton_gllP toggleButtonDisabled_aARS" type="button" disabled="" title="Switch between dark and light mode (currently light mode)" aria-label="Switch between dark and light mode (currently light mode)" aria-live="polite"><svg viewBox="0 0 24 24" width="24" height="24" class="lightToggleIcon_pyhR"><path fill="currentColor" d="M12,9c1.65,0,3,1.35,3,3s-1.35,3-3,3s-3-1.35-3-3S10.35,9,12,9 M12,7c-2.76,0-5,2.24-5,5s2.24,5,5,5s5-2.24,5-5 S14.76,7,12,7L12,7z M2,13l2,0c0.55,0,1-0.45,1-1s-0.45-1-1-1l-2,0c-0.55,0-1,0.45-1,1S1.45,13,2,13z M20,13l2,0c0.55,0,1-0.45,1-1 s-0.45-1-1-1l-2,0c-0.55,0-1,0.45-1,1S19.45,13,20,13z M11,2v2c0,0.55,0.45,1,1,1s1-0.45,1-1V2c0-0.55-0.45-1-1-1S11,1.45,11,2z M11,20v2c0,0.55,0.45,1,1,1s1-0.45,1-1v-2c0-0.55-0.45-1-1-1C11.45,19,11,19.45,11,20z M5.99,4.58c-0.39-0.39-1.03-0.39-1.41,0 c-0.39,0.39-0.39,1.03,0,1.41l1.06,1.06c0.39,0.39,1.03,0.39,1.41,0s0.39-1.03,0-1.41L5.99,4.58z M18.36,16.95 c-0.39-0.39-1.03-0.39-1.41,0c-0.39,0.39-0.39,1.03,0,1.41l1.06,1.06c0.39,0.39,1.03,0.39,1.41,0c0.39-0.39,0.39-1.03,0-1.41 L18.36,16.95z M19.42,5.99c0.39-0.39,0.39-1.03,0-1.41c-0.39-0.39-1.03-0.39-1.41,0l-1.06,1.06c-0.39,0.39-0.39,1.03,0,1.41 s1.03,0.39,1.41,0L19.42,5.99z M7.05,18.36c0.39-0.39,0.39-1.03,0-1.41c-0.39-0.39-1.03-0.39-1.41,0l-1.06,1.06 c-0.39,0.39-0.39,1.03,0,1.41s1.03,0.39,1.41,0L7.05,18.36z"></path></svg><svg viewBox="0 0 24 24" width="24" height="24" class="darkToggleIcon_wfgR"><path fill="currentColor" d="M9.37,5.51C9.19,6.15,9.1,6.82,9.1,7.5c0,4.08,3.32,7.4,7.4,7.4c0.68,0,1.35-0.09,1.99-0.27C17.45,17.19,14.93,19,12,19 c-3.86,0-7-3.14-7-7C5,9.07,6.81,6.55,9.37,5.51z M12,3c-4.97,0-9,4.03-9,9s4.03,9,9,9s9-4.03,9-9c0-0.46-0.04-0.92-0.1-1.36 c-0.98,1.37-2.58,2.26-4.4,2.26c-2.98,0-5.4-2.42-5.4-5.4c0-1.81,0.89-3.42,2.26-4.4C12.92,3.04,12.46,3,12,3L12,3z"></path></svg></button></div><div class="searchBox_ZlJk"></div></div></div><div role="presentation" class="navbar-sidebar__backdrop"></div></nav><div id="docusaurus_skipToContent_fallback" class="main-wrapper mainWrapper_z2l0 docsWrapper_BCFX"><button aria-label="Scroll back to top" class="clean-btn theme-back-to-top-button backToTopButton_sjWU" type="button"></button><div class="docPage__5DB"><aside class="theme-doc-sidebar-container docSidebarContainer_b6E3"><div class="sidebarViewport_Xe31"><div class="sidebar_njMd"><nav aria-label="Docs sidebar" class="menu thin-scrollbar menu_SIkG"><ul class="theme-doc-sidebar-menu menu__list"><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-1 menu__list-item"><a class="menu__link" href="/labtutorial/General/">Introduction</a></li><li class="theme-doc-sidebar-item-category theme-doc-sidebar-item-category-level-1 menu__list-item menu__list-item--collapsed"><div class="menu__list-item-collapsible"><a class="menu__link menu__link--sublist" aria-expanded="false" href="/labtutorial/General/FigureMaking/">Figure Making</a><button aria-label="Toggle the collapsible sidebar category &#x27;Figure Making&#x27;" type="button" class="clean-btn menu__caret"></button></div></li><li class="theme-doc-sidebar-item-category theme-doc-sidebar-item-category-level-1 menu__list-item"><div class="menu__list-item-collapsible"><a class="menu__link menu__link--sublist menu__link--active" aria-expanded="true" href="/labtutorial/General/PublicationChecklist/">Publication Checklist</a><button aria-label="Toggle the collapsible sidebar category &#x27;Publication Checklist&#x27;" type="button" class="clean-btn menu__caret"></button></div><ul style="display:block;overflow:visible;height:auto" class="menu__list"><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link" tabindex="0" href="/labtutorial/General/PublicationChecklist/pegr-projects">PEGR Projects</a></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link" tabindex="0" href="/labtutorial/General/PublicationChecklist/github">Github repo</a></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link menu__link--active" aria-current="page" tabindex="0" href="/labtutorial/General/PublicationChecklist/geo-submission">GEO Submission</a></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link" tabindex="0" href="/labtutorial/General/PublicationChecklist/zenodo">Zenodo Submission</a></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link" tabindex="0" href="/labtutorial/General/PublicationChecklist/stencil">STENCIL</a></li></ul></li></ul></nav></div></div></aside><main class="docMainContainer_gTbr"><div class="container padding-top--md padding-bottom--lg"><div class="row"><div class="col docItemCol_VOVn"><div class="docItemContainer_Djhp"><article><nav class="theme-doc-breadcrumbs breadcrumbsContainer_Z_bl" aria-label="Breadcrumbs"><ul class="breadcrumbs" itemscope="" itemtype="https://schema.org/BreadcrumbList"><li itemscope="" itemprop="itemListElement" itemtype="https://schema.org/ListItem" class="breadcrumbs__item"><a class="breadcrumbs__link" itemprop="item" href="/labtutorial/General/PublicationChecklist/"><span itemprop="name">Publication Checklist</span></a><meta itemprop="position" content="1"></li><li itemscope="" itemprop="itemListElement" itemtype="https://schema.org/ListItem" class="breadcrumbs__item breadcrumbs__item--active"><span class="breadcrumbs__link" itemprop="name">GEO Submission</span><meta itemprop="position" content="2"></li></ul></nav><div class="tocCollapsible_ETCw theme-doc-toc-mobile tocMobile_ITEo"><button type="button" class="clean-btn tocCollapsibleButton_TO0P">On this page</button></div><div class="theme-doc-markdown markdown"><header><h1>GEO Submission</h1></header><p>All data not subject to privacy restrictions (e.g. patient-protected samples) should be made publicly available at one of the national genomic data repositories. We submit through <a href="https://www.ncbi.nlm.nih.gov/geo/" target="_blank" rel="noopener noreferrer">Gene Expression Omnibus (GEO)</a>.</p><div class="theme-admonition theme-admonition-danger alert alert--danger admonition_LlT9"><div class="admonitionHeading_tbUL"><span class="admonitionIcon_kALy"><svg viewBox="0 0 12 16"><path fill-rule="evenodd" d="M5.05.31c.81 2.17.41 3.38-.52 4.31C3.55 5.67 1.98 6.45.9 7.98c-1.45 2.05-1.7 6.53 3.53 7.7-2.2-1.16-2.67-4.52-.3-6.61-.61 2.03.53 3.33 1.94 2.86 1.39-.47 2.3.53 2.27 1.67-.02.78-.31 1.44-1.13 1.81 3.42-.59 4.78-3.42 4.78-5.56 0-2.84-2.53-3.22-1.25-5.61-1.52.13-2.03 1.13-1.89 2.75.09 1.08-1.02 1.8-1.86 1.33-.67-.41-.66-1.19-.06-1.78C8.18 5.31 8.68 2.45 5.05.32L5.03.3l.02.01z"></path></svg></span>danger</div><div class="admonitionContent_S0QG"><p>DO NOT PROCEED unless you are ready to submit your manuscript. It is a pain in the butt to edit a submission later so it is better to finalize your datasets beforehand.</p></div></div><p>When the data used for your manuscript is somewhat finalized, follow the instructions here to submit your data. Allow for a couple days time for curators at NCBI to process samples but they usually have a quick turnaround time.</p><h2 class="anchor anchorWithStickyNavbar_LWe7" id="account-set-up">Account Set-up<a href="#account-set-up" class="hash-link" aria-label="Direct link to Account Set-up" title="Direct link to Account Set-up">​</a></h2><p>You will need a MyNCBI account if you don&#x27;t already have one. It should be linked to your ORCiD or Institution account. Make sure Frank&#x27;s information is included as your PI with all of the usual address and contact information.</p><h2 class="anchor anchorWithStickyNavbar_LWe7" id="get-the-latest-instructions">Get the latest instructions<a href="#get-the-latest-instructions" class="hash-link" aria-label="Direct link to Get the latest instructions" title="Direct link to Get the latest instructions">​</a></h2><p>Follow the <a href="https://www.ncbi.nlm.nih.gov/geo/info/submission.html" target="_blank" rel="noopener noreferrer">GEO submission instructions</a> for submitting the UNPUBLISHED datasets (&quot;Novel Set&quot;) used in the manuscript. Almost all data we produce are <strong>&quot;high-throughput sequencing data&quot;</strong> if they are coming from PEGR.</p><a href="https://www.ncbi.nlm.nih.gov/geo/info/seq.html" target="_blank" rel="noopener noreferrer"><div class="tutorial-img-flow-container"><img loading="lazy" src="/labtutorial/assets/images/GEO_SelectHTS-3fba2a58eb6d6477235aee4957217b18.png" style="width:100%" class="img_ev3q"></div></a><div class="theme-admonition theme-admonition-note alert alert--secondary admonition_LlT9"><div class="admonitionHeading_tbUL"><span class="admonitionIcon_kALy"><svg viewBox="0 0 14 16"><path fill-rule="evenodd" d="M6.3 5.69a.942.942 0 0 1-.28-.7c0-.28.09-.52.28-.7.19-.18.42-.28.7-.28.28 0 .52.09.7.28.18.19.28.42.28.7 0 .28-.09.52-.28.7a1 1 0 0 1-.7.3c-.28 0-.52-.11-.7-.3zM8 7.99c-.02-.25-.11-.48-.31-.69-.2-.19-.42-.3-.69-.31H6c-.27.02-.48.13-.69.31-.2.2-.3.44-.31.69h1v3c.02.27.11.5.31.69.2.2.42.31.69.31h1c.27 0 .48-.11.69-.31.2-.19.3-.42.31-.69H8V7.98v.01zM7 2.3c-3.14 0-5.7 2.54-5.7 5.68 0 3.14 2.56 5.7 5.7 5.7s5.7-2.55 5.7-5.7c0-3.15-2.56-5.69-5.7-5.69v.01zM7 .98c3.86 0 7 3.14 7 7s-3.14 7-7 7-7-3.12-7-7 3.14-7 7-7z"></path></svg></span>note</div><div class="admonitionContent_S0QG"><p>The instructions from GEO always supercede the documentation on this page. Please talk to a bioinformatician if you run into inconsistencies between the GEO instructions and this page.</p></div></div><h2 class="anchor anchorWithStickyNavbar_LWe7" id="get-your-list-of-pegr-sample-ids-to-submit">Get your list of PEGR sample IDs to submit<a href="#get-your-list-of-pegr-sample-ids-to-submit" class="hash-link" aria-label="Direct link to Get your list of PEGR sample IDs to submit" title="Direct link to Get your list of PEGR sample IDs to submit">​</a></h2><h3 class="anchor anchorWithStickyNavbar_LWe7" id="from-project-id">From Project ID<a href="#from-project-id" class="hash-link" aria-label="Direct link to From Project ID" title="Direct link to From Project ID">​</a></h3><p><a href="https://vesta.cac.cornell.edu/pegr/project/show/1331" target="_blank" rel="noopener noreferrer">1331</a> is the ID for <code>Krietenstein 2016 - Cell (Novel Set)</code>. Check the end of the project URL if you&#x27;re not sure you have the right project ID (vesta.cac.cornell.edu/pegr/project/show/<strong>1331</strong>)</p><h3 class="anchor anchorWithStickyNavbar_LWe7" id="from-sample-list">From Sample List<a href="#from-sample-list" class="hash-link" aria-label="Direct link to From Sample List" title="Direct link to From Sample List">​</a></h3><p>Or you can get a list of sample IDs by scrolling down to the &quot;Experiments&quot; section and clicking &quot;Export Samples to CSV&quot;</p><p><img loading="lazy" alt="PEGR_ExportSamplesFromProject" src="/labtutorial/assets/images/PEGR_ExportSamplesFromProject-080be9808d65a15ce815a61cd8c18882.png" width="708" height="290" class="img_ev3q"></p><div class="theme-admonition theme-admonition-caution alert alert--warning admonition_LlT9"><div class="admonitionHeading_tbUL"><span class="admonitionIcon_kALy"><svg viewBox="0 0 16 16"><path fill-rule="evenodd" d="M8.893 1.5c-.183-.31-.52-.5-.887-.5s-.703.19-.886.5L.138 13.499a.98.98 0 0 0 0 1.001c.193.31.53.501.886.501h13.964c.367 0 .704-.19.877-.5a1.03 1.03 0 0 0 .01-1.002L8.893 1.5zm.133 11.497H6.987v-2.003h2.039v2.003zm0-3.004H6.987V5.987h2.039v4.006z"></path></svg></span>caution</div><div class="admonitionContent_S0QG"><p>Just use the <a href="/labtutorial/General/PublicationChecklist/pegr-projects#novel-set">&quot;Novel set&quot;</a> samples. Previously published samples (<a href="/labtutorial/General/PublicationChecklist/pegr-projects#reanalyzed-set">&quot;Reanalyzed set&quot;</a>) that have been reanalyzed for this manuscript should NOT be included in the submission. These can be cited within the manuscript using their existing NCBI accession info.</p></div></div><h2 class="anchor anchorWithStickyNavbar_LWe7" id="retrieve-data-from-pegr">Retrieve data from PEGR<a href="#retrieve-data-from-pegr" class="hash-link" aria-label="Direct link to Retrieve data from PEGR" title="Direct link to Retrieve data from PEGR">​</a></h2><p>PEGR houses the metadata, raw data file links, and processed data file links that you need to create the GEO submission. We have built command line tools for facilitating the bulk retreival of this data.</p><div class="theme-admonition theme-admonition-danger alert alert--danger admonition_LlT9"><div class="admonitionHeading_tbUL"><span class="admonitionIcon_kALy"><svg viewBox="0 0 12 16"><path fill-rule="evenodd" d="M5.05.31c.81 2.17.41 3.38-.52 4.31C3.55 5.67 1.98 6.45.9 7.98c-1.45 2.05-1.7 6.53 3.53 7.7-2.2-1.16-2.67-4.52-.3-6.61-.61 2.03.53 3.33 1.94 2.86 1.39-.47 2.3.53 2.27 1.67-.02.78-.31 1.44-1.13 1.81 3.42-.59 4.78-3.42 4.78-5.56 0-2.84-2.53-3.22-1.25-5.61-1.52.13-2.03 1.13-1.89 2.75.09 1.08-1.02 1.8-1.86 1.33-.67-.41-.66-1.19-.06-1.78C8.18 5.31 8.68 2.45 5.05.32L5.03.3l.02.01z"></path></svg></span>danger</div><div class="admonitionContent_S0QG"><p>Please do not manually fill in the GEO submission form or manually download files as that has historically been mistake-prone. Use the PEGR API (see links below).</p></div></div><h3 class="anchor anchorWithStickyNavbar_LWe7" id="step-2-gather-raw-files-fastq">(Step 2) Gather raw files (FASTQ)<a href="#step-2-gather-raw-files-fastq" class="hash-link" aria-label="Direct link to (Step 2) Gather raw files (FASTQ)" title="Direct link to (Step 2) Gather raw files (FASTQ)">​</a></h3><p>Use <a href="https://github.com/CEGRcode/EGC_utility_scripts/blob/master/generate_FQ_file_from_PEGR.py" target="_blank" rel="noopener noreferrer">EGC_utility_scripts</a> to pull the raw FASTQ sample files</p><h3 class="anchor anchorWithStickyNavbar_LWe7" id="step-3-gather-processed-files">(Step 3) Gather processed files<a href="#step-3-gather-processed-files" class="hash-link" aria-label="Direct link to (Step 3) Gather processed files" title="Direct link to (Step 3) Gather processed files">​</a></h3><p>Samples may be &quot;processed&quot; differently depending on the data type. Use your judgement on what is appropriate for the purposes of the study. Consider referencing <a href="https://github.com/CEGRcode/2022-Mittal_SAGA/blob/main/03_Bulk_Processing/job/bulk_geo_tracks.pbs" target="_blank" rel="noopener noreferrer">this script</a> for creating BigWig pileups of 5&#x27; Read1 coverage across the genome. Or you can pull the BigWig files from the core pipeline (<code>paired_004</code>). If you are using <code>paired_004</code> BigWigs, note the computational steps from the pipeline and fill in the GEO submission accordingly.</p><h3 class="anchor anchorWithStickyNavbar_LWe7" id="step-4-gather-metadata">(Step 4) Gather metadata<a href="#step-4-gather-metadata" class="hash-link" aria-label="Direct link to (Step 4) Gather metadata" title="Direct link to (Step 4) Gather metadata">​</a></h3><p>Use <a href="https://github.com/CEGRcode/EGC_utility_scripts/blob/master/generate_Metadata_file_from_PEGR.py" target="_blank" rel="noopener noreferrer">EGC_utility_scripts</a> for pulling metadata from PEGR as a starting point for building the spreadsheet. <strong>You may need to manually adjust the metadata output of this script to fit GEO specifications for the submission.</strong></p><p>See GEO for downloading the metadata template on the instructions page. Fill it in with the sample metadata and manuscript information.</p><div class="theme-admonition theme-admonition-tip alert alert--success admonition_LlT9"><div class="admonitionHeading_tbUL"><span class="admonitionIcon_kALy"><svg viewBox="0 0 12 16"><path fill-rule="evenodd" d="M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"></path></svg></span>tip</div><div class="admonitionContent_S0QG"><p>Check Box for previous GEO submission spreadsheets.</p></div></div><h2 class="anchor anchorWithStickyNavbar_LWe7" id="step-5-get-md5-checksums-for-all-raw-and-processed-files">(Step 5) Get MD5 Checksums for all raw and processed files<a href="#step-5-get-md5-checksums-for-all-raw-and-processed-files" class="hash-link" aria-label="Direct link to (Step 5) Get MD5 Checksums for all raw and processed files" title="Direct link to (Step 5) Get MD5 Checksums for all raw and processed files">​</a></h2><p>You can use <a href="https://pughlab.mbg.cornell.edu/scriptmanager-docs/docs/Tools/file-utilities/md5checksum" target="_blank" rel="noopener noreferrer">ScriptManager</a> to do this for all your FASTQ and processed (e.g. BigWig) files.</p><h2 class="anchor anchorWithStickyNavbar_LWe7" id="step-6-organize-files-according-to-geo-instructions">(Step 6) Organize files according to GEO instructions<a href="#step-6-organize-files-according-to-geo-instructions" class="hash-link" aria-label="Direct link to (Step 6) Organize files according to GEO instructions" title="Direct link to (Step 6) Organize files according to GEO instructions">​</a></h2><p>Follow the GEO instructions for organizing your data. Please separate data from different assays as appropriate. For example, separate RNA-seq from ChIP-exo or ATAC-seq data.</p><h2 class="anchor anchorWithStickyNavbar_LWe7" id="step-7-upload-files-to-geo-servers">(Step 7) Upload files to GEO servers<a href="#step-7-upload-files-to-geo-servers" class="hash-link" aria-label="Direct link to (Step 7) Upload files to GEO servers" title="Direct link to (Step 7) Upload files to GEO servers">​</a></h2><img loading="lazy" src="/labtutorial/assets/images/GEO_UploadYourSubmission-d445811ecdf238b78a4bef973241a564.png" style="width:100%" class="img_ev3q"><h3 class="anchor anchorWithStickyNavbar_LWe7" id="1-transfer-files">1. Transfer files<a href="#1-transfer-files" class="hash-link" aria-label="Direct link to 1. Transfer files" title="Direct link to 1. Transfer files">​</a></h3><p>Then upload the data to their servers. They give you a bunch of options for uploading these data files. If you aren&#x27;t comfortable with FTP, try the FileZilla method (they have installation instructions if you don&#x27;t already have it).</p><img loading="lazy" src="/labtutorial/assets/images/GEO_FTPOptions-810307653ff0d7a72fa46d98c384078a.png" style="width:100%" class="img_ev3q"><h3 class="anchor anchorWithStickyNavbar_LWe7" id="2-submit-metadata">2. Submit metadata<a href="#2-submit-metadata" class="hash-link" aria-label="Direct link to 2. Submit metadata" title="Direct link to 2. Submit metadata">​</a></h3><p>Select the folder name that you transfered and upload the excel file you prepared.</p><div class="theme-admonition theme-admonition-caution alert alert--warning admonition_LlT9"><div class="admonitionHeading_tbUL"><span class="admonitionIcon_kALy"><svg viewBox="0 0 16 16"><path fill-rule="evenodd" d="M8.893 1.5c-.183-.31-.52-.5-.887-.5s-.703.19-.886.5L.138 13.499a.98.98 0 0 0 0 1.001c.193.31.53.501.886.501h13.964c.367 0 .704-.19.877-.5a1.03 1.03 0 0 0 .01-1.002L8.893 1.5zm.133 11.497H6.987v-2.003h2.039v2.003zm0-3.004H6.987V5.987h2.039v4.006z"></path></svg></span>caution</div><div class="admonitionContent_S0QG"><p>Regarding the date to make the data public, <strong>start with at least a year out</strong>. This can be changed later.</p></div></div><h2 class="anchor anchorWithStickyNavbar_LWe7" id="follow-up-on-the-submission">Follow up on the submission<a href="#follow-up-on-the-submission" class="hash-link" aria-label="Direct link to Follow up on the submission" title="Direct link to Follow up on the submission">​</a></h2><p>Once approved, you should get an email from GEO with the newly created <code>GSEXXXXXX</code> and <code>GSMXXXXXX</code> accessions to include in the manuscript.</p><ul><li>This typically means including the study (<code>GSEXXXXXX</code>) accession in the <strong>Methods</strong> or <strong>Data Access</strong> sections and the specific samples (<code>GSMXXXXXX</code>) in the <strong>extended methods/tables</strong>. The accessions of reanalyzed data from previously published samples should be similarly included.</li></ul><p>Also update with accession:</p><ul><li>Github repo</li><li>PEGR projects (Novel, reanalyzed, and working)</li></ul><h2 class="anchor anchorWithStickyNavbar_LWe7" id="after-manuscript-is-accepted">After manuscript is accepted...<a href="#after-manuscript-is-accepted" class="hash-link" aria-label="Direct link to After manuscript is accepted..." title="Direct link to After manuscript is accepted...">​</a></h2><p>Notify GEO so they can coordinate with the journal to release the data with the manuscript. We don&#x27;t want the manuscript to be in print but not have the data available to readers!</p></div><footer class="theme-doc-footer docusaurus-mt-lg"><div class="theme-doc-footer-edit-meta-row row"><div class="col"></div><div class="col lastUpdated_vwxv"><span class="theme-last-updated">Last updated<!-- --> on <b><time datetime="2024-05-16T18:47:01.000Z">May 16, 2024</time></b> by <b>Olivia Wen-Mei Lang</b></span></div></div></footer></article><nav class="pagination-nav docusaurus-mt-lg" aria-label="Docs pages navigation"><a class="pagination-nav__link pagination-nav__link--prev" href="/labtutorial/General/PublicationChecklist/github"><div class="pagination-nav__sublabel">Previous</div><div class="pagination-nav__label">Github repo</div></a><a class="pagination-nav__link pagination-nav__link--next" href="/labtutorial/General/PublicationChecklist/zenodo"><div class="pagination-nav__sublabel">Next</div><div class="pagination-nav__label">Zenodo Submission</div></a></nav></div></div><div class="col col--3"><div class="tableOfContents_bqdL thin-scrollbar theme-doc-toc-desktop"><ul class="table-of-contents table-of-contents__left-border"><li><a href="#account-set-up" class="table-of-contents__link toc-highlight">Account Set-up</a></li><li><a href="#get-the-latest-instructions" class="table-of-contents__link toc-highlight">Get the latest instructions</a></li><li><a href="#get-your-list-of-pegr-sample-ids-to-submit" class="table-of-contents__link toc-highlight">Get your list of PEGR sample IDs to submit</a><ul><li><a href="#from-project-id" class="table-of-contents__link toc-highlight">From Project ID</a></li><li><a href="#from-sample-list" class="table-of-contents__link toc-highlight">From Sample List</a></li></ul></li><li><a href="#retrieve-data-from-pegr" class="table-of-contents__link toc-highlight">Retrieve data from PEGR</a><ul><li><a href="#step-2-gather-raw-files-fastq" class="table-of-contents__link toc-highlight">(Step 2) Gather raw files (FASTQ)</a></li><li><a href="#step-3-gather-processed-files" class="table-of-contents__link toc-highlight">(Step 3) Gather processed files</a></li><li><a href="#step-4-gather-metadata" class="table-of-contents__link toc-highlight">(Step 4) Gather metadata</a></li></ul></li><li><a href="#step-5-get-md5-checksums-for-all-raw-and-processed-files" class="table-of-contents__link toc-highlight">(Step 5) Get MD5 Checksums for all raw and processed files</a></li><li><a href="#step-6-organize-files-according-to-geo-instructions" class="table-of-contents__link toc-highlight">(Step 6) Organize files according to GEO instructions</a></li><li><a href="#step-7-upload-files-to-geo-servers" class="table-of-contents__link toc-highlight">(Step 7) Upload files to GEO servers</a><ul><li><a href="#1-transfer-files" class="table-of-contents__link toc-highlight">1. Transfer files</a></li><li><a href="#2-submit-metadata" class="table-of-contents__link toc-highlight">2. Submit metadata</a></li></ul></li><li><a href="#follow-up-on-the-submission" class="table-of-contents__link toc-highlight">Follow up on the submission</a></li><li><a href="#after-manuscript-is-accepted" class="table-of-contents__link toc-highlight">After manuscript is accepted...</a></li></ul></div></div></div></div></main></div></div><footer class="footer footer--dark"><div class="container container-fluid"><div class="row footer__links"><div class="col footer__col"><div class="footer__title">Docs</div><ul class="footer__items clean-list"><li class="footer__item"><a class="footer__link-item" href="/labtutorial/General">Wiki Home</a></li></ul></div><div class="col footer__col"><div class="footer__title">Community</div><ul class="footer__items clean-list"><li class="footer__item"><a href="https://pughlab.mbg.cornell.edu" target="_blank" rel="noopener noreferrer" class="footer__link-item">Pugh Lab<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_nPIU"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></a></li><li class="footer__item"><a href="https://williamkmlai.github.io" target="_blank" rel="noopener noreferrer" class="footer__link-item">Lai Lab<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_nPIU"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></a></li></ul></div><div class="col footer__col"><div class="footer__title">Other Tools We Develop</div><ul class="footer__items clean-list"><li class="footer__item"><a href="http://pughlab.mbg.cornell.edu/scriptmanager-docs" target="_blank" rel="noopener noreferrer" class="footer__link-item">ScriptManager<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_nPIU"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></a></li><li class="footer__item"><a href="https://github.com/seqcode/pegr" target="_blank" rel="noopener noreferrer" class="footer__link-item">PEGR<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_nPIU"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></a></li><li class="footer__item"><a href="http://pughlab.mbg.cornell.edu/stencil" target="_blank" rel="noopener noreferrer" class="footer__link-item">STENCIL<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_nPIU"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></a></li><li class="footer__item"><a href="http://pughlab.mbg.cornell.edu/GenoPipe-docs" target="_blank" rel="noopener noreferrer" class="footer__link-item">GenoPipe<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_nPIU"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></a></li></ul></div></div><div class="footer__bottom text--center"><div class="margin-bottom--sm"><img src="/labtutorial/img/pughlab_logo.svg" alt="Pugh Lab Logo" class="themedImage_ToTc themedImage--light_HNdA footer__logo"><img src="/labtutorial/img/pughlab_logo.svg" alt="Pugh Lab Logo" class="themedImage_ToTc themedImage--dark_i4oU footer__logo"></div><div class="footer__copyright">Copyright © 2025 labtutorial. Built with Docusaurus.</div></div></div></footer></div>
<script src="/labtutorial/assets/js/runtime~main.78b75138.js"></script>
<script src="/labtutorial/assets/js/main.14ee8d51.js"></script>
</body>
</html>