<!DOCTYPE html>
<html lang="en">
<head>
<meta charset="utf-8">
<meta content="text/html; charset=utf-8" http-equiv="Content-Type">
<title>Galaxy - PughLab-guides</title>
<meta http-equiv="X-UA-Compatible" content="IE=edge">

<meta name="generator" content="mkdocs-1.1.2, mkdocs-gitbook-1.0.7">

<link rel="shortcut icon" href="../images/favicon.ico" type="image/x-icon">
<meta name="HandheldFriendly" content="true"/>
<meta name="viewport" content="width=device-width, initial-scale=1, user-scalable=no">
<meta name="apple-mobile-web-app-capable" content="yes">
<meta name="apple-mobile-web-app-status-bar-style" content="black">
<meta rel="next" href="" />
<link href="../css/style.min.css" rel="stylesheet"> 
</head>

<body>
<div class="book">
<div class="book-summary">

<nav role="navigation">
<ul class="summary">
<li>
<a href=".." target="_blank" class="custom-link">PughLab-guides</a>
</li>
<li class="divider"></li>
<li class="chapter" data-path="">
<a href="..">Home</a>
<li class="chapter active" data-path="galaxy/">
<a href="./">Galaxy</a>
<li class="chapter" data-path="planemo/">
<a href="../planemo/">Planemo</a>
<li class="chapter" data-path="CAC/">
<a href="../CAC/">CAC</a>
<li class="chapter" data-path="Frontend%20Deployment/">
<a href="../Frontend%20Deployment/">Frontend Deployment</a>
<li class="chapter" data-path="Backend%20Deployment/">
<a href="../Backend%20Deployment/">Backend Deployment</a>
<li class="chapter" data-path="pegr/">
<a href="../pegr/">PEGR</a>
<li class="chapter" data-path="FAQ/">
<a href="../FAQ/">FAQ</a>
<li class="divider"></li>



<li><a href="http://www.mkdocs.org">
Published with MkDocs
</a></li>

<li><a href="https://github.com/GitbookIO/theme-default">
Theme by GitBook
</a></li>
</ul>

</nav>

</div> <!-- end of book-summary -->

<div class="book-body">
<div class="body-inner">
<div class="book-header" role="navigation">

<!-- Title -->
<h1>
<i class="fa fa-circle-o-notch fa-spin"></i>
<a href="." ></a>
</h1>

</div> <!-- end of book-header -->

<div class="page-wrapper" tabindex="-1" role="main">
<div class="page-inner">

<section class="normal markdown-section">



<h2 id="dependencies">Dependencies<a class="headerlink" href="#dependencies" title="Permanent link">#</a></h2>
<ul>
<li>Assuming you have installed the required <a href="http://127.0.0.1:8000/#development-tools">development tools for Mac OS</a></li>
<li>Install <a href="http://samtools.sourceforge.net/-Install">samtools</a>.<ul>
<li><code>brew install samtools</code></li>
</ul>
</li>
<li>Install <a href="https://yarnpkg.com/en/docs/install">yarn</a>.<ul>
<li><code>brew install yarn</code></li>
</ul>
</li>
</ul>
<hr />
<h2 id="installing-galaxy">Installing Galaxy<a class="headerlink" href="#installing-galaxy" title="Permanent link">#</a></h2>
<p><em>It is recommended that you clone galaxy to your <code>~/Desktop</code>, so that GALAXY's internal paths are shorter and avoid conda throwing any errors that your path exceeds the maximum characters for tool install paths</em></p>
<p><em>here we are going to work with <code>17.09</code> release of GALAXY.</em></p>
<ul>
<li>Open a <code>terminal</code> and clone GALAXY to your <code>~/Desktop</code>.</li>
</ul>
<pre><code>cd ~/Desktop
git clone -b release_17.09 https://github.com/galaxyproject/galaxy.git
</code></pre>

<ul>
<li>Start GALAXY</li>
</ul>
<pre><code>cd galaxy
sh run.sh
</code></pre>

<p><em>Note that when you are starting galaxy for the first time, it takes some time to install required internal dependencies and setup its internal database. Once its gone through it, you should be able to see the start up page.</em></p>
<ul>
<li>Congratulations you have successfully installed GALAXY on your local machine.</li>
</ul>
<table>
<thead>
<tr>
<th>Useful resources &amp; tutorials</th>
</tr>
</thead>
<tbody>
<tr>
<td><a href="https://galaxyproject.org/admin/get-galaxy/">Get galaxy</a></td>
</tr>
<tr>
<td><a href="https://galaxyproject.org/learn/">Learn galaxy</a></td>
</tr>
<tr>
<td><a href="https://github.com/galaxyproject/dagobah-training">Dagobah training</a></td>
</tr>
</tbody>
</table>
<hr />
<h2 id="configuring-galaxy">Configuring Galaxy<a class="headerlink" href="#configuring-galaxy" title="Permanent link">#</a></h2>
<p><strong>Setting up an Admin</strong></p>
<p><code>galaxy.ini</code> file contains most of the configurations to your local galaxy. It has some predefined defaults which you might want to re-think before changing anything.</p>
<p><em>We will be using <code>cegr@psu.edu</code> as the default admin, so that it can be used to integrate with PEGR in the PEGR tutorial, where we use our local galaxy to send out information to PEGR</em></p>
<ul>
<li>Make sure your Galaxy is not online, before making any changes to configuration.</li>
<li>Open your galaxy folder in <code>Finder</code> on your mac. you should see a <code>config/</code> folder.</li>
<li>Inside the <code>config/</code>, you will find a <code>galaxy.ini.sample</code> file.</li>
<li>Copy the <code>galaxy.ini.sample</code> within the same folder and rename the copied file to <code>galaxy.ini</code></li>
<li>Open the <code>galaxy.ini</code> in a text editor of your choice.</li>
<li>Search for the line that starts with <code>admin_users</code> and add <code>cegr@psu.edu</code></li>
</ul>
<p><img alt="Galaxy Admin config Image" src="../image/localgalaxy_23.png" /></p>
<p><em>Above image shows the change. make sure you have edited <code>~/Desktop/galaxy/config/galaxy.ini</code></em></p>
<ul>
<li>Start your Galaxy now, <code>sh run.sh</code></li>
<li>Click on <code>User</code> menu and then click <code>register</code></li>
<li>You should see a registration page as below</li>
</ul>
<p><img alt="Galaxy register page" src="../image/register.png" /></p>
<ul>
<li>Enter <code>cegr@psu.edu</code> as the email address and choose a <code>password</code>, <code>public name</code> of your choice.</li>
<li>Once you login, you should now have the <code>Admin</code> menu show up on the menu bar on the top.</li>
</ul>
<p><img alt="Galaxy Admin option" src="../image/adminmenu.png" /></p>
<ul>
<li>When you click on the <code>Admin</code> tab, you should see the below page.</li>
</ul>
<p><img alt="Galaxy Admin page" src="../image/adminpage.png" /></p>
<hr />
<h2 id="adding-custom-genomes">Adding Custom Genomes<a class="headerlink" href="#adding-custom-genomes" title="Permanent link">#</a></h2>
<p><em><code>sacCer3_cegr</code> is the customized yeast genome we are using within the lab. Though the differences are few (to the best of my knowledge), but are very important to keep in mind while performing general data analysis. This genome deviates from the UCSC recommendations</em></p>
<p><em><code>sacCer3_cegr</code> contains <code>2-micron</code> regions &amp; chromosome naming is using <code>decimal number</code> instead of <code>roman numerals.</code> This causes some disadvantages to use tools like bedGraphToBigWig, etc.</em></p>
<ul>
<li>Click on <code>Admin</code> tab, then under <code>Tools and Tool Shed</code> section, click on <code>Search Tool Shed</code></li>
</ul>
<p><img alt="Galaxy ToolShed" src="../image/galaxy_toolshed.png" /></p>
<ul>
<li>Click on <code>Galaxy Main Tool Shed</code></li>
</ul>
<p><img alt="Galaxy ToolShed Categories" src="../image/toolshed_categories.png" /></p>
<ul>
<li>Search &amp; Install below tools. you can copy and paste below tool names into the search box.</li>
</ul>
<table>
<thead>
<tr>
<th>Tool Name</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>data_manager_bwa_mem_index_builder</code></td>
</tr>
<tr>
<td><code>data_manager_fetch_genome_dbkeys_all_fasta</code></td>
</tr>
<tr>
<td><code>data_manager_sam_fasta_index_builder</code></td>
</tr>
<tr>
<td><code>data_manager_twobit_builder</code></td>
</tr>
</tbody>
</table>
<ul>
<li>Below images previews the steps for installing a tool.</li>
</ul>
<p><img alt="Tool Install ex1" src="../image/install_ex1.png" /></p>
<p><img alt="Tool Install ex2" src="../image/install_ex2.png" /></p>
<p><img alt="Tool Install ex3" src="../image/install_ex3.png" /></p>
<ul>
<li>Once you have installed all the above tools, You can verify the installation at <code>Admin &gt; Tools and Tool shed &gt; Manage installed tools</code></li>
</ul>
<p><img alt="Manage installed tools" src="../image/localgalaxy_1.png" /></p>
<ul>
<li>Download the <code>sacCer3_cegr</code> genome from <a href="../image/data/sacCer3_cegr.fa">here</a></li>
<li>Upload the file you downloaded above into galaxy, using the <code>upload button</code> located on the tools menu, as shown in the image below.</li>
</ul>
<p><img alt="Upload button location" src="../image/upload_button.png" /></p>
<ul>
<li>Once you have uploaded, you should see the file appear in the “history pane” on your right.</li>
<li>Go to <code>Admin</code> tab, under <code>Data</code> section, Click on <code>Local data</code> to get <code>Data manager</code></li>
</ul>
<p><img alt="Data Manager" src="../image/localgalaxy_2.png" /></p>
<ul>
<li>We need to run below tools one after the other in the exact order mentioned below:</li>
</ul>
<table>
<thead>
<tr>
<th>Tool</th>
<th>Order</th>
</tr>
</thead>
<tbody>
<tr>
<td>Create DBkey and reference geneome</td>
<td>#1</td>
</tr>
<tr>
<td>TwoBit</td>
<td>#2</td>
</tr>
<tr>
<td>BWA-MEM</td>
<td>#3</td>
</tr>
<tr>
<td>SAM Fasta</td>
<td>#4</td>
</tr>
</tbody>
</table>
<p><em>All these tools are available from the <code>Admin &gt; Data &gt; Local data</code> section. Below are images for step-by-step execution of above tools in the same order. use them to fill out any default information that is required and follow along</em></p>
<ul>
<li>You can specify sequence name to be <code>sacCer3_cegr</code> and leave everything as default in all the tools.</li>
</ul>
<p><img alt="Step 1" src="../image/localgalaxy_3.png" />
<img alt="Step 2" src="../image/localgalaxy_4.png" />
<img alt="Step 3" src="../image/localgalaxy_5.png" />
<img alt="Step 4" src="../image/localgalaxy_6.png" />
<img alt="Step 5" src="../image/localgalaxy_7.png" /></p>
<ul>
<li>Once you have run all the tools, you need to check couple of things that populate in the internal database of galaxy. You can check that information from your <code>Admin</code> page.</li>
<li>Go to <code>Admin &gt; Data &gt; Local data</code> section, under <code>View Tool Data Table Entries</code>. Click on <code>__dbkeys__</code> you should see something like below.</li>
</ul>
<p><img alt="viewtools data table entries" src="../image/localgalaxy_11.png" /></p>
<p><img alt="Step 6" src="../image/localgalaxy_8.png" /></p>
<ul>
<li>Similarly check <code>all_fasta</code>,<code>twobit</code>, <code>bwa_mem_indexes</code>. All of them should contain an entry for <code>sacCer3_cegr</code>. Path could be different in your case.</li>
</ul>
<p><img alt="Step 7" src="../image/localgalaxy_9.png" />
<img alt="Step 8" src="../image/localgalaxy_10.png" />
<img alt="Step 9" src="../image/localgalaxy_12.png" /></p>
<ul>
<li>If your able to see similar results as above images. <code>Congratulations!</code> you have successfully added a custom genome build into your galaxy.</li>
</ul>
<hr />
<h2 id="importing-chip-exo-workflow">Importing ChIP-exo Workflow<a class="headerlink" href="#importing-chip-exo-workflow" title="Permanent link">#</a></h2>
<p><em>We will install the core-sequencing workflow that the lab uses to analyze all the samples that are sequenced. This pipeline is run to create BAM files, peak calling using genetrack and MEME motif analysis.</em></p>
<ul>
<li>Download the workflow file <a href="../image/data/Galaxy-Workflow-paired_002.ga">here.</a></li>
<li>Once you have downloaded the workflow. You can <code>import</code> it into your local galaxy from the <code>Workflow</code> tab using the <code>upload or import workflow</code> button located beside the search bar. <em>(see image below).</em></li>
</ul>
<p><img alt="Workflow upload" src="../image/localgalaxy_16.png" /></p>
<p><img alt="Workflow import menu" src="../image/localgalaxy_17.png" /></p>
<ul>
<li>Once you have selected the workflow file and clicked <code>import</code>. You might see some errors such as below. Nothing to worry, the error messages is caused by tools that are not yet installed &amp; are important for the workflow to run in your galaxy.</li>
</ul>
<p><img alt="Workflow import errors" src="../image/localgalaxy_18.png" /></p>
<ul>
<li>Click on <code>edit</code> option under the workflow drop-down menu. To find out the missing tools.</li>
<li>You need to install each tool manually from toolshed. Go to <code>Admin &gt; Tools and Tool Shed &gt; Search toolShed</code> to search and install each tool that is missing.</li>
</ul>
<p><em>Few tools have their toolnames in the workflow that end with <code>output_statistics</code>. These are the tools that are not available on <code>Galaxy toolshed</code> and need to be side-loaded separately, which we will do in the next section. so for now, you can ignore installing these tools and their errors.</em></p>
<hr />
<h2 id="integrating-cegr-output_statistics">Integrating CEGR <em><code>output_statistics</code></em><a class="headerlink" href="#integrating-cegr-output_statistics" title="Permanent link">#</a></h2>
<p><em>This section is similar to adding custom tools into Galaxy, here is a  (<a href="https://galaxyproject.org/admin/tools/add-tool-tutorial/">tutorial</a>). <code>cegr-galaxy</code> repo contains other important scripts that are used to run the core-sequencing pipeline on production galaxy. The repo has a <code>README</code> file detailing the usage of each script.</em></p>
<ul>
<li>
<p>Clone the repository containing the CEGR tools from <a href="https://github.com/seqcode/cegr-galaxy">seqcode/cegr-galaxy</a></p>
<ul>
<li><code>git clone https://github.com/seqcode/cegr-galaxy.git</code></li>
</ul>
</li>
<li>
<p>Copy the entire <code>cegr_statistics</code> folder to this location <code>galaxy/tools/</code> within your local galaxy.</p>
</li>
<li>Open <code>galaxy/config/tool_conf.xml</code> in a text editor of your choice.</li>
</ul>
<p><em>If the above file doesn't exist, there should be a file in the same config directory called <code>tool_conf.xml.main</code>, copy and rename the file to <code>tool_conf.xml</code></em></p>
<ul>
<li>Add below lines at the end of file, within the <code>&lt;/toolbox&gt;</code> tag.</li>
</ul>
<pre><code>&lt;section id=&quot;cegr_tools&quot; name=&quot;CEGR&quot;&gt;
    &lt;tool file=&quot;cegr_statistics/bam_to_scidx_output_stats.xml&quot; /&gt;
    &lt;tool file=&quot;cegr_statistics/bedtools_intersectbed_output_stats.xml&quot; /&gt;
    &lt;tool file=&quot;cegr_statistics/bwa_mem_output_stats_single.xml&quot; /&gt;
    &lt;tool file=&quot;cegr_statistics/cwpair2_output_stats.xml&quot; /&gt;
    &lt;tool file=&quot;cegr_statistics/extract_genomic_dna_output_stats.xml&quot; /&gt;
    &lt;tool file=&quot;cegr_statistics/extract_genomic_dna_output_stats2.xml&quot; /&gt;
    &lt;tool file=&quot;cegr_statistics/extract_genomic_dna_output_stats3.xml&quot; /&gt;
    &lt;tool file=&quot;cegr_statistics/fasta_nucleotide_color_plot_output_stats.xml&quot; /&gt;
    &lt;tool file=&quot;cegr_statistics/fastqc_output_stats.xml&quot; /&gt;
    &lt;tool file=&quot;cegr_statistics/fastqc_output_stats2.xml&quot; /&gt;
    &lt;tool file=&quot;cegr_statistics/genetrack_output_stats.xml&quot; /&gt;
    &lt;tool file=&quot;cegr_statistics/input_dataset_r1_output_stats.xml&quot; /&gt;
    &lt;tool file=&quot;cegr_statistics/input_dataset_r2_output_stats.xml&quot; /&gt;
    &lt;tool file=&quot;cegr_statistics/mark_duplicates_bam_output_stats.xml&quot; /&gt;
    &lt;tool file=&quot;cegr_statistics/meme_fimo_output_stats.xml&quot; /&gt;
    &lt;tool file=&quot;cegr_statistics/meme_meme_output_stats.xml&quot; /&gt;
    &lt;tool file=&quot;cegr_statistics/pe_histogram_output_stats.xml&quot; /&gt;
    &lt;tool file=&quot;cegr_statistics/repeatmasker_wrapper_output_stats.xml&quot; /&gt;
    &lt;tool file=&quot;cegr_statistics/repeatmasker_wrapper_output_stats2.xml&quot; /&gt;
    &lt;tool file=&quot;cegr_statistics/samtool_filter2_output_stats.xml&quot; /&gt;
    &lt;tool file=&quot;cegr_statistics/tag_pileup_frequency_output_stats.xml&quot; /&gt;
  &lt;/section&gt;

</code></pre>

<p><em>The above lines informs GALAXY, where it can find the tools and corresponding toolwrappers</em></p>
<ul>
<li>The file <code>galaxy/config/tool_config.xml</code> should look something like below:</li>
</ul>
<p><img alt="Adding CEGR tools" src="../image/addingcegrtools.png" /></p>
<ul>
<li>Save the file and restart Galaxy. You should now see these tools appear under <code>Tools</code> menu within galaxy's <code>Analyze Data</code> tab similar to below.</li>
</ul>
<p><img alt="CEGR tools view" src="../image/cegr_tools.png" /></p>
<ul>
<li><code>Congratulations!</code> you have successfully installed <code>output_statistics</code> tools into your galaxy.</li>
</ul>
<hr />
<h2 id="connecting-galaxy-to-pegr">Connecting Galaxy to PEGR<a class="headerlink" href="#connecting-galaxy-to-pegr" title="Permanent link">#</a></h2>
<p><em>Galaxy and PEGR communicate with each other using API keys. If you have not setup a local developmental PEGR. Set it up using these <a href="../pegr/">instructions</a> &amp; come back to this section</em></p>
<p><em>In the above section we installed <code>output_statistics</code> tools. These are the tools that send back information to <code>PEGR</code>. We did not configure the tools in the above section, which we will do in this section.</em></p>
<p><strong>Changing Galaxy's port</strong></p>
<ul>
<li>First, we will configure <code>Galaxy</code> to run on a different <code>port</code> so that it does not conflict with <code>PEGR</code>'s default port.</li>
<li>Open <code>galaxy/config/galaxy.ini</code> in a text editor and search for <code>port</code> setting. Change the port to <code>8090</code>. Below image shows the change in <code>galaxy.ini</code>.</li>
</ul>
<p><img alt="galaxy port change" src="../image/galaxy_port_change.png" /></p>
<p><strong>Generating Galaxy API key</strong></p>
<ul>
<li>Click on <code>Admin</code> tab, under <code>User Management</code> section, click on <code>API keys</code>.</li>
<li>Click on <code>Generate</code> button to create an API key. Don't regenerate a new key, if you already have one at this point.</li>
</ul>
<p><strong>CEGR <code>output_statistics</code> configuration</strong></p>
<ul>
<li>Open <code>galaxy/tools/stats_config.ini.sample</code> and add below information</li>
</ul>
<pre><code class="ini">
# Configuration file for the CEGR Galaxy ChIP-exo statistics tools.

[defaults]

# This section contains default settings for command line parameters that
# can be overridden when they are passed to executed scripts.

PEGR_API_KEY =  &lt;REPLACE THIS WITH PEGR API KEY&gt;
PEGR_URL = http://localhost:8080/pegr/api/stats

GALAXY_API_KEY = &lt;REPLACE THIS WITH YOUR GALAXY ADMIN USER's API KEY&gt;
GALAXY_BASE_URL = http://localhost:8090


[tool_categories]

input_dataset_r1 = output_fastqRead1
input_dataset_r2 = output_fastqRead2
toolshed.g2.bx.psu.edu/repos/iuc/bam_to_scidx/bam_to_scidx/1.0.1 = output_bamToScidx
toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_intersectbed/2.27.0.0 = output_bedtoolsIntersect
toolshed.g2.bx.psu.edu/repos/devteam/bwa/bwa_mem/0.7.17.1 = output_bwaMem
toolshed.g2.bx.psu.edu/repos/iuc/cwpair2/cwpair2/1.1.0 = output_cwpair2
toolshed.g2.bx.psu.edu/repos/iuc/genetrack/genetrack/1.0.1 = output_genetrack
toolshed.g2.bx.psu.edu/repos/iuc/extract_genomic_dna/Extract genomic DNA 1/3.0.3 = output_extractGenomicDNA
toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.70 = output_fastqc
toolshed.g2.bx.psu.edu/repos/iuc/pe_histogram/pe_histogram/1.0.1 = output_peHistogram
toolshed.g2.bx.psu.edu/repos/bgruening/repeat_masker/repeatmasker_wrapper/0.1.2 =output_repeatMasker
toolshed.g2.bx.psu.edu/repos/iuc/meme_meme/meme_meme/4.11.2.0 = output_meme
toolshed.g2.bx.psu.edu/repos/iuc/fasta_nucleotide_color_plot/fasta_nucleotide_color_plot/1.0.1 = output_fourColorPlot
toolshed.g2.bx.psu.edu/repos/jjohnson/samtools_filter/samtools_filter/1.1.1 = output_samtoolFilter
toolshed.g2.bx.psu.edu/repos/devteam/picard/picard_MarkDuplicates/2.7.1.1 = output_markDuplicates
toolshed.g2.bx.psu.edu/repos/iuc/meme_fimo/meme_fimo/4.11.2.0 = output_fimo
toolshed.g2.bx.psu.edu/repos/iuc/tag_pileup_frequency/tag_pileup_frequency/1.0.1 = output_tagPileup


</code></pre>

<ul>
<li>Below is an example configuration, after you add your API Keys.</li>
</ul>
<p><img alt="Example config" src="../image/sample_stats_config.png" /></p>
<ul>
<li>Rename <code>stats_config.ini.sample</code> to <code>stats_config.ini</code></li>
</ul>
<p><strong>Install bioblend</strong></p>
<p><em>bioblend is a python library to interact with Galaxy, using APIs</em></p>
<ul>
<li>Installation instructions <a href="https://bioblend.readthedocs.io/en/latest/#installation">here</a></li>
<li>If you have already installed <a href="https://www.anaconda.com/"><code>Anaconda</code></a> &amp; <a href="https://pypi.org/project/pip/"><code>pip</code></a><ul>
<li><code>pip install bioblend</code></li>
</ul>
</li>
<li>Using bioblend <a href="https://bioblend.readthedocs.io/en/latest/api_docs/galaxy/docs.html">docs</a></li>
</ul>
<p><strong>Adding core-sequencing workflowId into your local development PEGRdb</strong></p>
<p><em>This step is important, as it lets PEGR know which workflow stats to accept and update on PEGR web frontend</em></p>
<ul>
<li>Download <code>getWorkflowid.py</code> from <a href="../image/data/getWorkflowid.py">here</a></li>
<li>Open the script and replace the <code>url</code> and <code>key</code> with your local galaxy url and API key respectively. After adding your API-key, looks like the below image:</li>
</ul>
<p><img alt="Get workflowid" src="../image/getworkflowid.png" /></p>
<ul>
<li>Start your galaxy</li>
<li>Run the script to retrieve the workflowId. Below is an image showing expected output</li>
</ul>
<p><img alt="expected output for get workflowid" src="../image/localgalaxy_27.png" /></p>
<ul>
<li>Your workflow id is <code>ebfb8f50c6abde6d</code> in the above example.</li>
<li>Start your <code>MySQL</code> server and log into it using the username and password that you used while setting up <code>PEGR</code> or you can login as root  <code>mysql -u root -p</code></li>
</ul>
<pre><code>use pegr;
describe pipeline;
</code></pre>

<p><img alt="example query" src="../image/localgalaxy_28.png" /></p>
<ul>
<li>Edit below query with a <code>pipeline-name</code> of your choice and add the <code>workflowid</code> you retrieved from above script and execute it. Below is an example query that you need</li>
</ul>
<pre><code>insert into pipeline( version,name,note,pipeline_version,steps,workflow_id)
values( 1,

'&lt;pipeline-name&gt;',

'Locally installed galaxy sending JSON dictionaires to locally installed pegr','001', '[[&quot;input_dataset_r1_output_stats&quot;,&quot;fastqRead1&quot;],[&quot;input_dataset_r2_output_stats&quot;,&quot;fastqRead2&quot;],[&quot;fastqc_output_stats&quot;,&quot;fastqc&quot;],[&quot;fastqc_output_stats2&quot;,&quot;fastqc&quot;],[&quot;mark_duplicates_bam_output_stats&quot;,&quot;markDuplicates&quot;],[&quot;samtool_filter2_output_stats&quot;,&quot;samtoolFilter&quot;],[&quot;pe_histogram_output_stats&quot;,&quot;peHistogram&quot;],[&quot;bam_to_scidx_output_stats&quot;,&quot;bamToScidx&quot;],[&quot;genetrack_output_stats&quot;,&quot;genetrack&quot;],[&quot;bedtools_intersectbed_output_stats&quot;,&quot;bedtoolsIntersect&quot;],[&quot;cwpair2_output_stats&quot;,&quot;cwpair2&quot;],[&quot;extract_genomic_dna_output_stats&quot;,&quot;extractGenomicDNA&quot;],[&quot;extract_genomic_dna_output_stats2&quot;,&quot;extractGenomicDNA&quot;],[&quot;repeatmasker_wrapper_output_stats&quot;,&quot;repeatMasker&quot;],[&quot;repeatmasker_wrapper_output_stats2&quot;,&quot;repeatMasker&quot;],[&quot;meme_meme_output_stats&quot;,&quot;meme&quot;],[&quot;meme_fimo_output_stats&quot;,&quot;fimo&quot;],[&quot;extract_genomic_dna_output_stats3&quot;,&quot;extractGenomicDNA&quot;],[&quot;fasta_nucleotide_color_plot_output_stats&quot;,&quot;fourColorPlot&quot;],[&quot;tag_pileup_frequency_output_stats&quot;,&quot;tagPileup&quot;]]' ,

&quot;&lt;workflowid&gt;&quot;);

</code></pre>

<ul>
<li>This how your query looks like after adding your pipeline-name and workflowid</li>
</ul>
<p><img alt="query" src="../image/localgalaxy_29.png" /></p>
<ul>
<li>You are all set for executing the pipeline. <em>(if you have followed and set up the keys correctly, there should be no errors)</em></li>
</ul>


</section>

</div> <!-- end of page-inner -->
</div> <!-- end of page-wrapper -->

</div> <!-- end of body-inner -->

</div> <!-- end of book-body -->
<script src="../js/main.js"></script>
<script src="../js/gitbook.min.js"></script>
<script src="../js/theme.min.js"></script>
</body>
</html>