<!doctype html>
<html lang="en" dir="ltr" class="docs-wrapper docs-doc-page docs-version-current plugin-docs plugin-id-default docs-doc-id-Software/peakcalling-tutorial">
<head>
<meta charset="UTF-8">
<meta name="generator" content="Docusaurus v2.4.0">
<title data-rh="true">ChIP-exo Peak Calling Tutorial | Lab Wiki</title><meta data-rh="true" name="viewport" content="width=device-width,initial-scale=1"><meta data-rh="true" name="twitter:card" content="summary_large_image"><meta data-rh="true" property="og:url" content="https://CEGRcode.github.io/labtutorial/Software/peakcalling-tutorial"><meta data-rh="true" name="docusaurus_locale" content="en"><meta data-rh="true" name="docsearch:language" content="en"><meta data-rh="true" name="docusaurus_version" content="current"><meta data-rh="true" name="docusaurus_tag" content="docs-default-current"><meta data-rh="true" name="docsearch:version" content="current"><meta data-rh="true" name="docsearch:docusaurus_tag" content="docs-default-current"><meta data-rh="true" property="og:title" content="ChIP-exo Peak Calling Tutorial | Lab Wiki"><meta data-rh="true" name="description" content="Generate peak calling frequency plots from ChIP-exo data"><meta data-rh="true" property="og:description" content="Generate peak calling frequency plots from ChIP-exo data"><link data-rh="true" rel="icon" href="/labtutorial/img/favicon.png"><link data-rh="true" rel="canonical" href="https://CEGRcode.github.io/labtutorial/Software/peakcalling-tutorial"><link data-rh="true" rel="alternate" href="https://CEGRcode.github.io/labtutorial/Software/peakcalling-tutorial" hreflang="en"><link data-rh="true" rel="alternate" href="https://CEGRcode.github.io/labtutorial/Software/peakcalling-tutorial" hreflang="x-default"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.13.24/dist/katex.min.css" integrity="sha384-odtC+0UGzzFL/6PNoE8rX/SPcQDXBJ+uRepguP4QkPCm2LBxH3FA3y+fKSiJ+AmM" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.3.0/css/all.min.css" integrity="sha512-SzlrxWUlpfuzQ+pcUCosxcglQRNAq/DZjVsC0lE40xsADsfeQoEypE+enwcOiGjk/bSuGGKHEyjSoQ1zVisanQ==" crossorigin="anonymous"><link rel="stylesheet" href="/labtutorial/assets/css/styles.e7b83035.css">
<link rel="preload" href="/labtutorial/assets/js/runtime~main.bf2c6d3b.js" as="script">
<link rel="preload" href="/labtutorial/assets/js/main.928c6cc0.js" as="script">
</head>
<body class="navigation-with-keyboard">
<script>!function(){function t(t){document.documentElement.setAttribute("data-theme",t)}var e=function(){var t=null;try{t=new URLSearchParams(window.location.search).get("docusaurus-theme")}catch(t){}return t}()||function(){var t=null;try{t=localStorage.getItem("theme")}catch(t){}return t}();t(null!==e?e:"light")}()</script><div id="__docusaurus">
<div role="region" aria-label="Skip to main content"><a class="skipToContent_fXgn" href="#docusaurus_skipToContent_fallback">Skip to main content</a></div><nav aria-label="Main" class="navbar navbar--fixed-top"><div class="navbar__inner"><div class="navbar__items"><button aria-label="Toggle navigation bar" aria-expanded="false" class="navbar__toggle clean-btn" type="button"><svg width="30" height="30" viewBox="0 0 30 30" aria-hidden="true"><path stroke="currentColor" stroke-linecap="round" stroke-miterlimit="10" stroke-width="2" d="M4 7h22M4 15h22M4 23h22"></path></svg></button><a href="http://pughlab.mbg.cornell.edu/" target="_blank" rel="noopener noreferrer" class="navbar__brand"><div class="navbar__logo"><img src="/labtutorial/img/logo.svg" alt="Lab logo" class="themedImage_ToTc themedImage--light_HNdA"><img src="/labtutorial/img/logo_dark.svg" alt="Lab logo" class="themedImage_ToTc themedImage--dark_i4oU"></div></a><a class="navbar__item navbar__link" href="/labtutorial/General/">Publication Checklist</a><a aria-current="page" class="navbar__item navbar__link navbar__link--active" href="/labtutorial/Software/">Software</a><a class="navbar__item navbar__link" href="/labtutorial/WetBench/">Wet Bench</a></div><div class="navbar__items navbar__items--right"><a href="https://github.com/CEGRcode" target="_blank" rel="noopener noreferrer" class="navbar__item navbar__link header-github-link"></a><div class="toggle_vylO colorModeToggle_DEke"><button class="clean-btn toggleButton_gllP toggleButtonDisabled_aARS" type="button" disabled="" title="Switch between dark and light mode (currently light mode)" aria-label="Switch between dark and light mode (currently light mode)" aria-live="polite"><svg viewBox="0 0 24 24" width="24" height="24" class="lightToggleIcon_pyhR"><path fill="currentColor" d="M12,9c1.65,0,3,1.35,3,3s-1.35,3-3,3s-3-1.35-3-3S10.35,9,12,9 M12,7c-2.76,0-5,2.24-5,5s2.24,5,5,5s5-2.24,5-5 S14.76,7,12,7L12,7z M2,13l2,0c0.55,0,1-0.45,1-1s-0.45-1-1-1l-2,0c-0.55,0-1,0.45-1,1S1.45,13,2,13z M20,13l2,0c0.55,0,1-0.45,1-1 s-0.45-1-1-1l-2,0c-0.55,0-1,0.45-1,1S19.45,13,20,13z M11,2v2c0,0.55,0.45,1,1,1s1-0.45,1-1V2c0-0.55-0.45-1-1-1S11,1.45,11,2z M11,20v2c0,0.55,0.45,1,1,1s1-0.45,1-1v-2c0-0.55-0.45-1-1-1C11.45,19,11,19.45,11,20z M5.99,4.58c-0.39-0.39-1.03-0.39-1.41,0 c-0.39,0.39-0.39,1.03,0,1.41l1.06,1.06c0.39,0.39,1.03,0.39,1.41,0s0.39-1.03,0-1.41L5.99,4.58z M18.36,16.95 c-0.39-0.39-1.03-0.39-1.41,0c-0.39,0.39-0.39,1.03,0,1.41l1.06,1.06c0.39,0.39,1.03,0.39,1.41,0c0.39-0.39,0.39-1.03,0-1.41 L18.36,16.95z M19.42,5.99c0.39-0.39,0.39-1.03,0-1.41c-0.39-0.39-1.03-0.39-1.41,0l-1.06,1.06c-0.39,0.39-0.39,1.03,0,1.41 s1.03,0.39,1.41,0L19.42,5.99z M7.05,18.36c0.39-0.39,0.39-1.03,0-1.41c-0.39-0.39-1.03-0.39-1.41,0l-1.06,1.06 c-0.39,0.39-0.39,1.03,0,1.41s1.03,0.39,1.41,0L7.05,18.36z"></path></svg><svg viewBox="0 0 24 24" width="24" height="24" class="darkToggleIcon_wfgR"><path fill="currentColor" d="M9.37,5.51C9.19,6.15,9.1,6.82,9.1,7.5c0,4.08,3.32,7.4,7.4,7.4c0.68,0,1.35-0.09,1.99-0.27C17.45,17.19,14.93,19,12,19 c-3.86,0-7-3.14-7-7C5,9.07,6.81,6.55,9.37,5.51z M12,3c-4.97,0-9,4.03-9,9s4.03,9,9,9s9-4.03,9-9c0-0.46-0.04-0.92-0.1-1.36 c-0.98,1.37-2.58,2.26-4.4,2.26c-2.98,0-5.4-2.42-5.4-5.4c0-1.81,0.89-3.42,2.26-4.4C12.92,3.04,12.46,3,12,3L12,3z"></path></svg></button></div><div class="searchBox_ZlJk"></div></div></div><div role="presentation" class="navbar-sidebar__backdrop"></div></nav><div id="docusaurus_skipToContent_fallback" class="main-wrapper mainWrapper_z2l0 docsWrapper_BCFX"><button aria-label="Scroll back to top" class="clean-btn theme-back-to-top-button backToTopButton_sjWU" type="button"></button><div class="docPage__5DB"><aside class="theme-doc-sidebar-container docSidebarContainer_b6E3"><div class="sidebarViewport_Xe31"><div class="sidebar_njMd"><nav aria-label="Docs sidebar" class="menu thin-scrollbar menu_SIkG"><ul class="theme-doc-sidebar-menu menu__list"><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-1 menu__list-item"><a class="menu__link" href="/labtutorial/Software/">Introduction</a></li><li class="theme-doc-sidebar-item-category theme-doc-sidebar-item-category-level-1 menu__list-item menu__list-item--collapsed"><div class="menu__list-item-collapsible"><a class="menu__link menu__link--sublist" aria-expanded="false" href="/labtutorial/Software/NewMembers/">New Members</a><button aria-label="Toggle the collapsible sidebar category &#x27;New Members&#x27;" type="button" class="clean-btn menu__caret"></button></div></li><li class="theme-doc-sidebar-item-category theme-doc-sidebar-item-category-level-1 menu__list-item menu__list-item--collapsed"><div class="menu__list-item-collapsible"><a class="menu__link menu__link--sublist" aria-expanded="false" href="/labtutorial/Software/SetupGuides/">Set-up Guides</a><button aria-label="Toggle the collapsible sidebar category &#x27;Set-up Guides&#x27;" type="button" class="clean-btn menu__caret"></button></div></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-1 menu__list-item"><a class="menu__link" href="/labtutorial/Software/compute">Computing Resources</a></li><li class="theme-doc-sidebar-item-category theme-doc-sidebar-item-category-level-1 menu__list-item menu__list-item--collapsed"><div class="menu__list-item-collapsible"><a class="menu__link menu__link--sublist" aria-expanded="false" href="/labtutorial/Software/Websites/">Websites</a><button aria-label="Toggle the collapsible sidebar category &#x27;Websites&#x27;" type="button" class="clean-btn menu__caret"></button></div></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-1 menu__list-item"><a class="menu__link menu__link--active" aria-current="page" href="/labtutorial/Software/peakcalling-tutorial">ChIP-exo Peak Calling</a></li><li class="theme-doc-sidebar-item-category theme-doc-sidebar-item-category-level-1 menu__list-item menu__list-item--collapsed"><div class="menu__list-item-collapsible"><a class="menu__link menu__link--sublist" aria-expanded="false" href="/labtutorial/Software/UsingJetstream2/">Using Jetstream2</a><button aria-label="Toggle the collapsible sidebar category &#x27;Using Jetstream2&#x27;" type="button" class="clean-btn menu__caret"></button></div></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-1 menu__list-item"><a class="menu__link" href="/labtutorial/Software/cac">Cornell&#x27;s CAC server setup</a></li><li class="theme-doc-sidebar-item-category theme-doc-sidebar-item-category-level-1 menu__list-item menu__list-item--collapsed"><div class="menu__list-item-collapsible"><a class="menu__link menu__link--sublist" aria-expanded="false" href="/labtutorial/Software/PSU_Archive/">PSU Archive</a><button aria-label="Toggle the collapsible sidebar category &#x27;PSU Archive&#x27;" type="button" class="clean-btn menu__caret"></button></div></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-1 menu__list-item"><a class="menu__link" href="/labtutorial/Software/contributers">Contributors</a></li></ul></nav></div></div></aside><main class="docMainContainer_gTbr"><div class="container padding-top--md padding-bottom--lg"><div class="row"><div class="col docItemCol_VOVn"><div class="docItemContainer_Djhp"><article><nav class="theme-doc-breadcrumbs breadcrumbsContainer_Z_bl" aria-label="Breadcrumbs"><ul class="breadcrumbs" itemscope="" itemtype="https://schema.org/BreadcrumbList"><li itemscope="" itemprop="itemListElement" itemtype="https://schema.org/ListItem" class="breadcrumbs__item breadcrumbs__item--active"><span class="breadcrumbs__link" itemprop="name">ChIP-exo Peak Calling</span><meta itemprop="position" content="1"></li></ul></nav><div class="tocCollapsible_ETCw theme-doc-toc-mobile tocMobile_ITEo"><button type="button" class="clean-btn tocCollapsibleButton_TO0P">On this page</button></div><div class="theme-doc-markdown markdown"><header><h1>ChIP-exo Peak Calling Tutorial</h1></header><p><em>Generate peak calling frequency plots from ChIP-exo data</em></p><p><strong>Goal:</strong> This tutorial provides a guide to performing a peak calling analysis using the ScriptManager platform and data generated by the <a href="http://www.yeastepigenome.org/" target="_blank" rel="noopener noreferrer">Yeast Epigenome project</a>.</p><h2 class="anchor anchorWithStickyNavbar_LWe7" id="download-scriptmanager-v014">Download ScriptManager (v0.14):<a href="#download-scriptmanager-v014" class="hash-link" aria-label="Direct link to Download ScriptManager (v0.14):" title="Direct link to Download ScriptManager (v0.14):">​</a></h2><p>The current version of ScriptManager is available for download <strong><a href="https://github.com/CEGRcode/scriptmanager/releases/download/v0.14/ScriptManager-v0.14.jar" target="_blank" rel="noopener noreferrer">here</a></strong>. Make sure you have <a href="http://pughlab.mbg.cornell.edu/scriptmanager-docs/docs/#java" target="_blank" rel="noopener noreferrer">Java installed</a>.</p><p>The file <code>ScriptManager-v0.14.jar</code> should be placed someplace locally accessible. For example on Mac OS on the Desktop (Permissions will need to be accepted) or someplace in your home directory (<code>Macintosh_HD:Users/userID/ScriptManager</code>)</p><h2 class="anchor anchorWithStickyNavbar_LWe7" id="download-python-scripts">Download Python Scripts<a href="#download-python-scripts" class="hash-link" aria-label="Direct link to Download Python Scripts" title="Direct link to Download Python Scripts">​</a></h2><p>This tutorial will use Python scripts GeneTrack and CWPair. Make sure you have Python installed.</p><ol><li><a href="ftp://data1.commons.psu.edu/pub/commons/eberly/pughlab/yeast-epigenome-project/12141_YEP.zip" target="_blank" rel="noopener noreferrer" class="button button--secondary">Download GeneTrack script</a></li><li><a href="ftp://data1.commons.psu.edu/pub/commons/eberly/pughlab/yeast-epigenome-project/12141_YEP.zip" target="_blank" rel="noopener noreferrer" class="button button--secondary">Download CWPair script</a></li></ol><h2 class="anchor anchorWithStickyNavbar_LWe7" id="optional-download-uscs-binary-bedgraphtobigwig">(Optional) Download USCS binary <code>bedGraphToBigWig</code><a href="#optional-download-uscs-binary-bedgraphtobigwig" class="hash-link" aria-label="Direct link to optional-download-uscs-binary-bedgraphtobigwig" title="Direct link to optional-download-uscs-binary-bedgraphtobigwig">​</a></h2><p>You will need to download the precompiled binary for converting bedGraphs to BigWig files. Navigate to <a href="https://hgdownload.soe.ucsc.edu/admin/exe/" target="_blank" rel="noopener noreferrer">UCSC executables FTP</a>. Click the link to the OS that matches your machine and scroll down to download the <code>bedGraphToBigWig</code> binary.</p><div class="tabs-container tabList__CuJ"><ul role="tablist" aria-orientation="horizontal" class="tabs"><li role="tab" tabindex="0" aria-selected="true" class="tabs__item tabItem_LNqP tabs__item--active">Linux/MacOS</li></ul><div class="margin-top--md"><div role="tabpanel" class="tabItem_Ymn6"><p>Make sure you modify the permissions so that you can execute the binary. You can do this with <code>chmod</code></p><div class="codeBlockContainer_Ckt0 theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_biex"><pre tabindex="0" class="prism-code language-text codeBlock_bY9V thin-scrollbar"><code class="codeBlockLines_e6Vv"><span class="token-line" style="color:#393A34"><span class="token plain">chmod 755 /path/to/bedGraphToBigWig</span><br></span></code></pre><div class="buttonGroup__atx"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_eSgA" aria-hidden="true"><svg class="copyButtonIcon_y97N" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_LjdS" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>For example, if your <code>bedGraphToBigWig</code> binary downloaded to your <code>Downloads</code> folder,</p><div class="codeBlockContainer_Ckt0 theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_biex"><pre tabindex="0" class="prism-code language-text codeBlock_bY9V thin-scrollbar"><code class="codeBlockLines_e6Vv"><span class="token-line" style="color:#393A34"><span class="token plain">chmod 755 ~/Downloads/bedGraphToBigWig</span><br></span></code></pre><div class="buttonGroup__atx"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_eSgA" aria-hidden="true"><svg class="copyButtonIcon_y97N" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_LjdS" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div></div></div></div><h2 class="anchor anchorWithStickyNavbar_LWe7" id="download-data">Download Data<a href="#download-data" class="hash-link" aria-label="Direct link to Download Data" title="Direct link to Download Data">​</a></h2><p>You need one set of aligned DNA fragments (BAM) to complete this exercise. <a href="http://pughlab.mbg.cornell.edu/scriptmanager-docs/docs/References/file-formats" target="_blank" rel="noopener noreferrer">Read more about the BED/BAM file formats here.</a></p><h3 class="anchor anchorWithStickyNavbar_LWe7" id="bam-file">BAM File<a href="#bam-file" class="hash-link" aria-label="Direct link to BAM File" title="Direct link to BAM File">​</a></h3><p>This is the set of Reb1 read alignments from the Yeast Epigenome Project (YEP). <a href="https://pubmed.ncbi.nlm.nih.gov/33692541/" target="_blank" rel="noopener noreferrer">See Rossi et al (2021)</a> for more details.</p><div class="tutorial-img-flow-container"><img loading="lazy" src="/labtutorial/assets/images/download-yep-home-f975840806150a454e5ec6533fd7fd53.png" style="width:60%" class="img_ev3q"></div><ol><li>Navigate to <a href="http://www.yeastepigenome.org/" target="_blank" rel="noopener noreferrer">www.yeastepigenome.org</a> and search for Reb1</li><li>Select &quot;META DATA&quot;</li><li>Select &quot;Direct Download&quot;</li><li>Unzip the resulting file ‘12141_YEP.zip’ and inspect the contents of the new <code>12141_YEP</code> folder. It should contain a file called <code>12141_filtered.bam</code>.</li></ol><div class="tutorial-img-flow-container"><img loading="lazy" src="/labtutorial/assets/images/download-yep-reb1-3dd6db352f8dab08f27e1cad8092367f.png" style="width:50%" class="img_ev3q"><img loading="lazy" src="/labtutorial/assets/images/download-yep-reb1-metadata-ab2aeca9e2417d0c313723d71a34d85b.png" style="width:50%" class="img_ev3q"></div><h3 class="anchor anchorWithStickyNavbar_LWe7" id="optional-xxxxchromsizes-reference-file">(Optional) XXXX.chrom.sizes Reference File<a href="#optional-xxxxchromsizes-reference-file" class="hash-link" aria-label="Direct link to (Optional) XXXX.chrom.sizes Reference File" title="Direct link to (Optional) XXXX.chrom.sizes Reference File">​</a></h3><p>You will also need the chromosome sizes file of the yeast reference genome (sacCer3):</p><a href="https://hgdownload.soe.ucsc.edu/goldenPath/sacCer3/bigZips/sacCer3.chrom.sizes" target="_blank" rel="noopener noreferrer" class="button button--secondary">Download sacCer3.chrom.sizes (TXT)</a><br><br><div class="theme-admonition theme-admonition-warning alert alert--danger admonition_LlT9"><div class="admonitionHeading_tbUL"><span class="admonitionIcon_kALy"><svg viewBox="0 0 12 16"><path fill-rule="evenodd" d="M5.05.31c.81 2.17.41 3.38-.52 4.31C3.55 5.67 1.98 6.45.9 7.98c-1.45 2.05-1.7 6.53 3.53 7.7-2.2-1.16-2.67-4.52-.3-6.61-.61 2.03.53 3.33 1.94 2.86 1.39-.47 2.3.53 2.27 1.67-.02.78-.31 1.44-1.13 1.81 3.42-.59 4.78-3.42 4.78-5.56 0-2.84-2.53-3.22-1.25-5.61-1.52.13-2.03 1.13-1.89 2.75.09 1.08-1.02 1.8-1.86 1.33-.67-.41-.66-1.19-.06-1.78C8.18 5.31 8.68 2.45 5.05.32L5.03.3l.02.01z"></path></svg></span>danger</div><div class="admonitionContent_S0QG"><p>The downloaded sacCer3.chrom.sizes file may be using the roman numeral naming system (<code>chrI chrII chrIII ...</code>). You will need to modify this to the arabic (<code>chr1 chr2 chr3 ...</code>) numeral naming system if you plan to use the BAM file linked above. You can do this in a simple text editor.</p><p>Also, if the file is missing the <code>2-micron</code> row, you will need to add it manually. Simply append the following tab-delimited row to the bottom of the <code>sacCer3.chrom.sizes</code> file.</p><div class="codeBlockContainer_Ckt0 theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_biex"><pre tabindex="0" class="prism-code language-text codeBlock_bY9V thin-scrollbar"><code class="codeBlockLines_e6Vv"><span class="token-line" style="color:#393A34"><span class="token plain">2-micron    6318</span><br></span></code></pre><div class="buttonGroup__atx"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_eSgA" aria-hidden="true"><svg class="copyButtonIcon_y97N" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_LjdS" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div></div></div><h2 class="anchor anchorWithStickyNavbar_LWe7" id="convert-bam-file-to-scidx-format">Convert BAM File to scIDX Format<a href="#convert-bam-file-to-scidx-format" class="hash-link" aria-label="Direct link to Convert BAM File to scIDX Format" title="Direct link to Convert BAM File to scIDX Format">​</a></h2><h3 class="anchor anchorWithStickyNavbar_LWe7" id="1-open-scriptmanager">1. Open ScriptManager<a href="#1-open-scriptmanager" class="hash-link" aria-label="Direct link to 1. Open ScriptManager" title="Direct link to 1. Open ScriptManager">​</a></h3><div class="tabs-container tabList__CuJ"><ul role="tablist" aria-orientation="horizontal" class="tabs"><li role="tab" tabindex="0" aria-selected="true" class="tabs__item tabItem_LNqP tabs__item--active">MacOS</li><li role="tab" tabindex="-1" aria-selected="false" class="tabs__item tabItem_LNqP">Linux</li><li role="tab" tabindex="-1" aria-selected="false" class="tabs__item tabItem_LNqP">Windows</li></ul><div class="margin-top--md"><div role="tabpanel" class="tabItem_Ymn6"><p>Depending on your system permissions, you may need to be an administrator to open this for the first time. On Mac systems, this can be done by right-clicking the file and selecting ‘Open’ at the top.</p><div class="tutorial-img-flow-container"><img loading="lazy" src="/labtutorial/assets/images/MacOpenSM-5f0508d59e500792ea199946818706f1.png" style="width:40%;border:solid 1px" class="img_ev3q"></div><div class="theme-admonition theme-admonition-caution alert alert--warning admonition_LlT9"><div class="admonitionHeading_tbUL"><span class="admonitionIcon_kALy"><svg viewBox="0 0 16 16"><path fill-rule="evenodd" d="M8.893 1.5c-.183-.31-.52-.5-.887-.5s-.703.19-.886.5L.138 13.499a.98.98 0 0 0 0 1.001c.193.31.53.501.886.501h13.964c.367 0 .704-.19.877-.5a1.03 1.03 0 0 0 .01-1.002L8.893 1.5zm.133 11.497H6.987v-2.003h2.039v2.003zm0-3.004H6.987V5.987h2.039v4.006z"></path></svg></span>caution</div><div class="admonitionContent_S0QG"><p>Some MacOS systems may not properly open the JAR by simply double-clicking on the JAR file so you may need to open your Terminal window and execute it from the command line by executing the jar file <em>without any arguments or flags</em>:</p><div class="codeBlockContainer_Ckt0 theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_biex"><pre tabindex="0" class="prism-code language-text codeBlock_bY9V thin-scrollbar"><code class="codeBlockLines_e6Vv"><span class="token-line" style="color:#393A34"><span class="token plain">java -jar /path/to/ScriptManager.jar</span><br></span></code></pre><div class="buttonGroup__atx"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_eSgA" aria-hidden="true"><svg class="copyButtonIcon_y97N" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_LjdS" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>If you&#x27;re not sure how to type the path to ScriptManager, you can type <code>java -jar </code> (end with space) and then drag ScriptManager from Finder into your Terminal window and then press enter.</p></div></div></div><div role="tabpanel" class="tabItem_Ymn6" hidden=""><p>Double-click or right-click the ScriptManager JAR file to start the program.</p><div class="tutorial-img-flow-container"><img loading="lazy" src="/labtutorial/assets/images/LinuxOpenSM-1fe13f9ab9c369dd24c0a67c7ee86f41.png" style="width:40%;border:solid 1px" class="img_ev3q"></div></div><div role="tabpanel" class="tabItem_Ymn6" hidden=""><p>Double-click or right-click the ScriptManager JAR file to start the program.</p><div class="tutorial-img-flow-container"><img loading="lazy" src="/labtutorial/assets/images/WindowsOpenSM-ac32d3855e9f0cf632ab34c5c9fc983d.png" style="width:40%;border:solid 1px" class="img_ev3q"></div></div></div></div><p>Once you see the main tool selection window, you&#x27;re off to the races!</p><h3 class="anchor anchorWithStickyNavbar_LWe7" id="2-generate-bai-index-file">2. Generate BAI index file<a href="#2-generate-bai-index-file" class="hash-link" aria-label="Direct link to 2. Generate BAI index file" title="Direct link to 2. Generate BAI index file">​</a></h3><p>A BAI index file is required for each BAM file of interest (i.e., the tag occupancy data you want to plot). This file allows for rapid access of the sorted and aligned sequence reads (BAM file).</p><div class="theme-admonition theme-admonition-tip alert alert--success admonition_LlT9"><div class="admonitionHeading_tbUL"><span class="admonitionIcon_kALy"><svg viewBox="0 0 12 16"><path fill-rule="evenodd" d="M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"></path></svg></span>tip</div><div class="admonitionContent_S0QG"><p>SAM/BAM standard is to keep BAI file in same directory as BAM file with the ScriptManager-generated filename.</p><div class="language-bash codeBlockContainer_Ckt0 theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_biex"><pre tabindex="0" class="prism-code language-bash codeBlock_bY9V thin-scrollbar"><code class="codeBlockLines_e6Vv"><span class="token-line" style="color:#393A34"><span class="token plain">Reb1_YEP_12141.bam</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">Reb1_YEP_12141.bam.bai  </span><span class="token comment" style="color:#999988;font-style:italic"># Need to generate this file to proceed.</span><br></span></code></pre><div class="buttonGroup__atx"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_eSgA" aria-hidden="true"><svg class="copyButtonIcon_y97N" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_LjdS" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div></div></div><h4 class="anchor anchorWithStickyNavbar_LWe7" id="21-navigate-to-bam-manipulation-️-bam-bai-indexer">2.1. Navigate to BAM Manipulation ➡️ <a href="http://pughlab.mbg.cornell.edu/scriptmanager-docs/docs/bam-manipulation/bam-indexer" target="_blank" rel="noopener noreferrer"><strong>BAM-BAI Indexer</strong></a><a href="#21-navigate-to-bam-manipulation-️-bam-bai-indexer" class="hash-link" aria-label="Direct link to 21-navigate-to-bam-manipulation-️-bam-bai-indexer" title="Direct link to 21-navigate-to-bam-manipulation-️-bam-bai-indexer">​</a></h4><div class="tutorial-img-flow-container"><img loading="lazy" src="/labtutorial/assets/images/maingui-bamidx-916174344aa9670cd3fe1bb3f0287a74.png" style="width:60%" class="img_ev3q"></div><h4 class="anchor anchorWithStickyNavbar_LWe7" id="22-generate-bai-index-files-for-each-bam-file-of-interest-by-loading-your-bam-file-and-clicking-index">2.2. Generate BAI index files for each BAM file of interest by loading your BAM file and clicking &quot;Index.&quot;<a href="#22-generate-bai-index-files-for-each-bam-file-of-interest-by-loading-your-bam-file-and-clicking-index" class="hash-link" aria-label="Direct link to 2.2. Generate BAI index files for each BAM file of interest by loading your BAM file and clicking &quot;Index.&quot;" title="Direct link to 2.2. Generate BAI index files for each BAM file of interest by loading your BAM file and clicking &quot;Index.&quot;">​</a></h4><div class="tutorial-img-flow-container"><img loading="lazy" src="/labtutorial/assets/images/gui-bamidx-aa4e45d9e6ea63a6f3bb4a921bfd1ad5.png" style="width:60%" class="img_ev3q"><i class="fa-solid fa-arrow-right fa-lg"></i><img loading="lazy" src="/labtutorial/assets/images/gui-bamidx-complete-e925a25e686f6c6813848dd104f8cfd5.png" style="width:30%" class="img_ev3q"></div><div class="theme-admonition theme-admonition-note alert alert--secondary admonition_LlT9"><div class="admonitionHeading_tbUL"><span class="admonitionIcon_kALy"><svg viewBox="0 0 14 16"><path fill-rule="evenodd" d="M6.3 5.69a.942.942 0 0 1-.28-.7c0-.28.09-.52.28-.7.19-.18.42-.28.7-.28.28 0 .52.09.7.28.18.19.28.42.28.7 0 .28-.09.52-.28.7a1 1 0 0 1-.7.3c-.28 0-.52-.11-.7-.3zM8 7.99c-.02-.25-.11-.48-.31-.69-.2-.19-.42-.3-.69-.31H6c-.27.02-.48.13-.69.31-.2.2-.3.44-.31.69h1v3c.02.27.11.5.31.69.2.2.42.31.69.31h1c.27 0 .48-.11.69-.31.2-.19.3-.42.31-.69H8V7.98v.01zM7 2.3c-3.14 0-5.7 2.54-5.7 5.68 0 3.14 2.56 5.7 5.7 5.7s5.7-2.55 5.7-5.7c0-3.15-2.56-5.69-5.7-5.69v.01zM7 .98c3.86 0 7 3.14 7 7s-3.14 7-7 7-7-3.12-7-7 3.14-7 7-7z"></path></svg></span>note</div><div class="admonitionContent_S0QG"><p>The speed of this step scales with the size of the BAM file. Generally this step 30 sec for a 100 MB BAM file but may take 1-2 min for a multi-GB BAM file.</p></div></div><h3 class="anchor anchorWithStickyNavbar_LWe7" id="3-create-a-whole-genome-pileup-using-bam-format-converter-tools">3. Create a whole-genome pileup using BAM Format Converter Tools<a href="#3-create-a-whole-genome-pileup-using-bam-format-converter-tools" class="hash-link" aria-label="Direct link to 3. Create a whole-genome pileup using BAM Format Converter Tools" title="Direct link to 3. Create a whole-genome pileup using BAM Format Converter Tools">​</a></h3><h4 class="anchor anchorWithStickyNavbar_LWe7" id="21-navigate-to-bam-format-converter-️-bam-to-scidx-to-do-a-genome-wide-pileup-of-your-bam">2.1. Navigate to BAM Format Converter ➡️ <a href="http://pughlab.mbg.cornell.edu/scriptmanager-docs/docs/bam-format-converter/bam-to-scidx" target="_blank" rel="noopener noreferrer"><strong>BAM to scIDX</strong></a> to do a genome-wide pileup of your BAM<a href="#21-navigate-to-bam-format-converter-️-bam-to-scidx-to-do-a-genome-wide-pileup-of-your-bam" class="hash-link" aria-label="Direct link to 21-navigate-to-bam-format-converter-️-bam-to-scidx-to-do-a-genome-wide-pileup-of-your-bam" title="Direct link to 21-navigate-to-bam-format-converter-️-bam-to-scidx-to-do-a-genome-wide-pileup-of-your-bam">​</a></h4><div class="tutorial-img-flow-container"><img loading="lazy" src="/labtutorial/assets/images/maingui-bamtoscidx-673594bf8e06d0b7074206cda67ec2fb.png" style="width:60%" class="img_ev3q"></div><h4 class="anchor anchorWithStickyNavbar_LWe7" id="22-select-encoding-information">2.2. Select Encoding information<a href="#22-select-encoding-information" class="hash-link" aria-label="Direct link to 2.2. Select Encoding information" title="Direct link to 2.2. Select Encoding information">​</a></h4><p>Since this is a ChIP-exo dataset, we will select Read 1 (5&#x27; end) for the encoding. You may select insert size filters if you wish.</p><h4 class="anchor anchorWithStickyNavbar_LWe7" id="23-execute-pileup">2.3. Execute pileup<a href="#23-execute-pileup" class="hash-link" aria-label="Direct link to 2.3. Execute pileup" title="Direct link to 2.3. Execute pileup">​</a></h4><div class="tutorial-img-flow-container"><img loading="lazy" src="/labtutorial/assets/images/gui-bamtoscidx-ad3c2db72cce5b6733b0e39637c7a997.png" style="width:60%" class="img_ev3q"><i class="fa-solid fa-arrow-right fa-lg"></i><img loading="lazy" src="/labtutorial/assets/images/gui-bamtoscidx-output-ee4852e4db0dbbcab067e03bfe1f8537.png" style="width:40%" class="img_ev3q"></div><p>This should output a <code>12141_filtered_READ1.tab</code> file.</p><h2 class="anchor anchorWithStickyNavbar_LWe7" id="genetrack-script">GeneTrack Script<a href="#genetrack-script" class="hash-link" aria-label="Direct link to GeneTrack Script" title="Direct link to GeneTrack Script">​</a></h2><p>Open the terminal on your computer. Navigate to the directory with the <code>genetrack.py</code> file and run the following command:</p><div class="codeBlockContainer_Ckt0 theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_biex"><pre tabindex="0" class="prism-code language-text codeBlock_bY9V thin-scrollbar"><code class="codeBlockLines_e6Vv"><span class="token-line" style="color:#393A34"><span class="token plain">python genetrack.py /path/to/12141_filtered_READ1.tab</span><br></span></code></pre><div class="buttonGroup__atx"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_eSgA" aria-hidden="true"><svg class="copyButtonIcon_y97N" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_LjdS" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p><strong>Output called peaks in GFF format</strong></p><div class="tutorial-img-flow-container"><img loading="lazy" src="/labtutorial/assets/images/genetrack_output-97fc98d9d00e6e8454cbae460ddb1825.png" style="width:60%" class="img_ev3q"></div><p>This should output a folder labeled <code>genetrack_s5e20F1</code> in the directory of your scIDX file. Inside of the folder, the <code>12141_filtered_READ1_s5e20F1.gff</code> file will be used for the CWPair script.</p><p>The GeneTrack script supports multiple input files. You can run the script using a file or list of files, a directory or list of directories, or use &quot;.&quot; to run the script in the current directory.</p><h3 class="anchor anchorWithStickyNavbar_LWe7" id="output-options">Output Options<a href="#output-options" class="hash-link" aria-label="Direct link to Output Options" title="Direct link to Output Options">​</a></h3><table><thead><tr><th>Option</th><th>Description</th></tr></thead><tbody><tr><td><code>-s=&lt;SIGMA&gt;</code></td><td>Sigma to use when smoothing reads to call peaks (default=5)</td></tr><tr><td><code>-e=&lt;EXCLUSION&gt;</code></td><td>Exclusion zone around each peak that prevents others from being called (default=20)</td></tr><tr><td><code>-u=&lt;UP_WIDTH&gt;</code></td><td>Upstream width of called peaks (default=half exclusion zone)</td></tr><tr><td><code>-d=&lt;DOWN_WIDTH&gt;</code></td><td>Downstream width of called peaks (default=half exclusion zone)</td></tr><tr><td><code>-F=&lt;FILTER&gt;</code></td><td>Absolute read filter; outputs only peaks with larger read count (default=1)</td></tr></tbody></table><h2 class="anchor anchorWithStickyNavbar_LWe7" id="cwpair-script">CWPair Script<a href="#cwpair-script" class="hash-link" aria-label="Direct link to CWPair Script" title="Direct link to CWPair Script">​</a></h2><p>Now that the GeneTrack script output has been generated, we can run the CWPair script. Navigate to the directory with the <code>cwpair.py</code> file and run the following command:</p><div class="codeBlockContainer_Ckt0 theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_biex"><pre tabindex="0" class="prism-code language-text codeBlock_bY9V thin-scrollbar"><code class="codeBlockLines_e6Vv"><span class="token-line" style="color:#393A34"><span class="token plain">python cwpair.py /path/to/12141_filtered_READ1_s5e20F1.gff</span><br></span></code></pre><div class="buttonGroup__atx"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_eSgA" aria-hidden="true"><svg class="copyButtonIcon_y97N" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_LjdS" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>This should output a folder labeled <code>cwpair_output_mode_f0u50d100b1</code> inside of the previous <code>genetrack_s5e20F1</code> folder. This folder contains the following files:</p><p><code>S_12141_filtered_READ1_s5e20F1.gff</code> (simple),</p><p><code>D_12141_filtered_READ1_s5e20F1.txt</code> (detailed),</p><p><code>O_12141_filtered_READ1_s5e20F1.txt</code> (orphans),</p><p><code>P_12141_filtered_READ1_s5e20F1.pdf</code> (frequency preview plot),</p><p><code>F_12141_filtered_READ1_s5e20F1.pdf</code> (final frequency plot),</p><p><code>statistics.txt</code> (summarizes each input file)</p><div class="tutorial-img-flow-container"><img loading="lazy" src="/labtutorial/assets/images/cwpair_output_P-235620bace82f24c7e465084158830d1.png" style="width:50%" class="img_ev3q"><i class="fa-solid fa-arrow-right fa-lg"></i><img loading="lazy" src="/labtutorial/assets/images/cwpair_output_F-c501742fbd4e57bb343929f5f1089355.png" style="width:50%" class="img_ev3q"></div><p>.</p><p>The CWPair script supports multiple input files. You can run the script using a file or list of file, a directory or list of directories, or use &quot;.&quot; to run the script in the current directory.</p><h3 class="anchor anchorWithStickyNavbar_LWe7" id="output-options-1">Output Options<a href="#output-options-1" class="hash-link" aria-label="Direct link to Output Options" title="Direct link to Output Options">​</a></h3><table><thead><tr><th>Option</th><th>Description</th></tr></thead><tbody><tr><td><code>-m=&lt;METHOD&gt;</code></td><td>Method of finding match. Valid options are mode (default), closest, largest, and all (run with each method)</td></tr><tr><td><code>-u=&lt;UP_DISTANCE&gt;</code></td><td>Distance upstream of a Watson peak to allow a Crick pair (default=50)</td></tr><tr><td><code>-d=&lt;DOWN_DISTANCE&gt;</code></td><td>Distance downstream of a Watson peak to allow a Crick pair (default=100)</td></tr><tr><td><code>-d=&lt;BINSIZE&gt;</code></td><td>Width of bins for frequency plots and mode calculation (default=1)</td></tr><tr><td><code>-F=&lt;ABSOLUTE_THRESHOLD&gt;</code></td><td>Absolute read filter; outputs only peaks with larger read count (default=0)</td></tr><tr><td><code>-p=&lt;PLOT_FORMAT&gt;</code></td><td>Format to output graph in. Valid options are png, svg, and pdf (default)</td></tr><tr><td><code>-o=&lt;FORMAT&gt;</code></td><td>Format to output simple output file in. Valid options are gff (default) and txt</td></tr><tr><td><code>-c=&lt;SORT_CHROMOSOME&gt;</code></td><td>Output files will be sorted by chromosome. Valid options are asc (default) and desc</td></tr><tr><td><code>-s=&lt;SORT_SCORE&gt;</code></td><td>Output files will be sorted by score. Valid option are asc, desc, and none (default)</td></tr></tbody></table><h2 class="anchor anchorWithStickyNavbar_LWe7" id="optional-view-pileup-in-genome-browser">(Optional) View Pileup in Genome Browser<a href="#optional-view-pileup-in-genome-browser" class="hash-link" aria-label="Direct link to (Optional) View Pileup in Genome Browser" title="Direct link to (Optional) View Pileup in Genome Browser">​</a></h2><p>Follow these steps to turn your BAM files into genome track files that can be viewed in a genome browser!</p><h3 class="anchor anchorWithStickyNavbar_LWe7" id="convert-scidx-to-bigwig">Convert scIDX to BigWig<a href="#convert-scidx-to-bigwig" class="hash-link" aria-label="Direct link to Convert scIDX to BigWig" title="Direct link to Convert scIDX to BigWig">​</a></h3><p>Since ScriptManager currently does not support converting BedGraph files to BigWigs, this step must be done in your terminal/on the command-prompt.</p><h4 class="anchor anchorWithStickyNavbar_LWe7" id="1-turn-your-scidx-file-into-a-bedgraph-file">1. Turn your scIDX file into a bedGraph file<a href="#1-turn-your-scidx-file-into-a-bedgraph-file" class="hash-link" aria-label="Direct link to 1. Turn your scIDX file into a bedGraph file" title="Direct link to 1. Turn your scIDX file into a bedGraph file">​</a></h4><div class="tabs-container tabList__CuJ"><ul role="tablist" aria-orientation="horizontal" class="tabs"><li role="tab" tabindex="0" aria-selected="true" class="tabs__item tabItem_LNqP tabs__item--active">Shell</li><li role="tab" tabindex="-1" aria-selected="false" class="tabs__item tabItem_LNqP">Excel</li></ul><div class="margin-top--md"><div role="tabpanel" class="tabItem_Ymn6"><div class="codeBlockContainer_Ckt0 theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_biex"><pre tabindex="0" class="prism-code language-text codeBlock_bY9V thin-scrollbar"><code class="codeBlockLines_e6Vv"><span class="token-line" style="color:#393A34"><span class="token plain">grep -v -f sacCer3.chrom.sizes 12141_filtered_READ1_SCALE.tab &gt; 12141_filtered_READ1_SCALE.filtered.tab</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">sed &#x27;1d;2d&#x27; 12141_filtered_READ1_SCALE.filtered.tab | awk &#x27;{OFS=&quot;\t&quot;}{FS=&quot;\t&quot;}{print $1, $2, $2+1, $3}&#x27; |sort -k1,1 -k2,2n &gt; 12141_filtered_READ1_SCALE_forward.bedgraph</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">sed &#x27;1d;2d&#x27; 12141_filtered_READ1_SCALE.filtered.tab | awk &#x27;{OFS=&quot;\t&quot;}{FS=&quot;\t&quot;}{print $1, $2, $2+1, $4}&#x27; |sort -k1,1 -k2,2n &gt; 12141_filtered_READ1_SCALE_reverse.bedgraph</span><br></span></code></pre><div class="buttonGroup__atx"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_eSgA" aria-hidden="true"><svg class="copyButtonIcon_y97N" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_LjdS" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div></div><div role="tabpanel" class="tabItem_Ymn6" hidden=""><p>Open your file in excel and strip the file of the first two rows:</p><div class="codeBlockContainer_Ckt0 theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_biex"><pre tabindex="0" class="prism-code language-text codeBlock_bY9V thin-scrollbar"><code class="codeBlockLines_e6Vv"><span class="token-line" style="color:#393A34"><span class="token plain">#2023-03-10 12:52:40.439;12141_filtered.bam;READ1</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">chrom   index   forward reverse value</span><br></span></code></pre><div class="buttonGroup__atx"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_eSgA" aria-hidden="true"><svg class="copyButtonIcon_y97N" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_LjdS" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>You may also have read mapping to the very edges of the chromosome. These entries will also have to be removed.</p><p>Then rearrange the columns into BedGraph format by copying them in the following order (3rd column is the value of col2 plus one):</p><ul><li><code>12141_filtered_READ1_SCALE_forward.bedGraph</code>: col1, col2, (col2 +1), col3</li><li><code>12141_filtered_READ1_SCALE_reverse.bedGraph</code>: col1, col2, (col2 +1), col4</li></ul><p>...and then sort each of them by the first and then the second columns. This format is tab-delimited (not comma-delimited!) so save as a tab-delimited text file. Do not save this as an Excel file format (<code>.xlsx</code>, <code>.xls</code>, etc). This should be a <code>.tsv</code> or <code>.tab</code> format.</p></div></div></div><h4 class="anchor anchorWithStickyNavbar_LWe7" id="2-turn-your-bedgraph-file-into-a-bigwig-file">2. Turn your bedGraph file into a BigWig file<a href="#2-turn-your-bedgraph-file-into-a-bigwig-file" class="hash-link" aria-label="Direct link to 2. Turn your bedGraph file into a BigWig file" title="Direct link to 2. Turn your bedGraph file into a BigWig file">​</a></h4><div class="tabs-container tabList__CuJ"><ul role="tablist" aria-orientation="horizontal" class="tabs"><li role="tab" tabindex="0" aria-selected="true" class="tabs__item tabItem_LNqP tabs__item--active">Shell</li></ul><div class="margin-top--md"><div role="tabpanel" class="tabItem_Ymn6"><div class="codeBlockContainer_Ckt0 theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_biex"><pre tabindex="0" class="prism-code language-text codeBlock_bY9V thin-scrollbar"><code class="codeBlockLines_e6Vv"><span class="token-line" style="color:#393A34"><span class="token plain">./bedGraphToBigWig 12141_filtered_READ1_SCALE_forward.bedGraph sacCer3.chrom.sizes my-track-basename_forward.bw</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">./bedGraphToBigWig 12141_filtered_READ1_SCALE_reverse.bedGraph sacCer3.chrom.sizes my-track-basename_reverse.bw</span><br></span></code></pre><div class="buttonGroup__atx"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_eSgA" aria-hidden="true"><svg class="copyButtonIcon_y97N" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_LjdS" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div></div></div></div><h4 class="anchor anchorWithStickyNavbar_LWe7" id="3-load-the-bigwig-into-your-favorite-genome-browser">3. Load the BigWig into your <a href="https://software.broadinstitute.org/software/igv/download" target="_blank" rel="noopener noreferrer">favorite genome browser</a><a href="#3-load-the-bigwig-into-your-favorite-genome-browser" class="hash-link" aria-label="Direct link to 3-load-the-bigwig-into-your-favorite-genome-browser" title="Direct link to 3-load-the-bigwig-into-your-favorite-genome-browser">​</a></h4><div class="tutorial-img-flow-container"><img loading="lazy" src="/labtutorial/assets/images/IGV_BrowserShot-1ff8f8b15497181fcb1800cc6fc746f5.png" style="width:80%" class="img_ev3q"></div><p>Tah dah! You&#x27;ve made the genome tracks for a ChIP-exo dataset!</p><h2 class="anchor anchorWithStickyNavbar_LWe7" id="command-line-shell-script">Command-Line shell script<a href="#command-line-shell-script" class="hash-link" aria-label="Direct link to Command-Line shell script" title="Direct link to Command-Line shell script">​</a></h2><div class="codeBlockContainer_Ckt0 theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_biex"><pre tabindex="0" class="prism-code language-text codeBlock_bY9V thin-scrollbar"><code class="codeBlockLines_e6Vv"><span class="token-line" style="color:#393A34"><span class="token plain">SCRIPTMANAGER=/path/to/ScriptManager.jar</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">BGTOBW=/path/to/bedGraphtoBigWig</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">BAMFILE=/path/to/12141_filtered.bam</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">OUTPUT=/path/to/output-basename</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">CHRMSZ=/path/to/sacCer3.chrom.sizes</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">GENETRACK=/path/to/genetrack.py</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">CWPAIR=/path/to/cwpair.py</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain" style="display:inline-block"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"># Index BAM if not already indexed</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">[ -f $BAMFILE.bai ] || samtools index $BAMFILE</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"># Pileup BAM along whole genome</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">java -jar $SCRIPTMANAGER bam-format-converter bam-to-scidx $BAMFILE -o $OUTPUT.tab</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain" style="display:inline-block"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"># Run GeneTrack script</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">python $GENETRACK $OUTPUT.tab</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain" style="display:inline-block"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"># Run CWPair script</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">python $CWPAIR $OUTPUT_s5e20F1.gff</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain" style="display:inline-block"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"># (Optional) View pileup in genome browser</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain" style="display:inline-block"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"># Only include pileups from chromosomes in the sacCer3.chrom.sizes reference</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">grep -v -f $CHRMSZ $OUTPUT.tab &gt; $OUTPUT.filtered.tab</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"># Convert to forward/reverse Bedgraph tracks</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">sed &#x27;1d;2d&#x27; $OUTPUT.scaled.tab | awk &#x27;{OFS=&quot;\t&quot;}{FS=&quot;\t&quot;}{print $1, $2, $2+1, $3}&#x27; |sort -k1,1 -k2,2n &gt; $OUTPUT.forward.bedgraph</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">sed &#x27;1d;2d&#x27; $OUTPUT.scaled.tab | awk &#x27;{OFS=&quot;\t&quot;}{FS=&quot;\t&quot;}{print $1, $2, $2+1, $4}&#x27; |sort -k1,1 -k2,2n &gt; $OUTPUT.reverse.bedgraph</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"># Compress to BigWig format</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">$BGTOBW $OUTPUT.forward.bedgraph $CHRMSZ $OUTPUT\_forward.bw</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">$BGTOBW $OUTPUT.reverse.bedgraph $CHRMSZ $OUTPUT\_reverse.bw</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain" style="display:inline-block"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"># Clean-up</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">rm $OUTPUT.tab $OUTPUT.filtered.tab $OUTPUT.forward.bedgraph $OUTPUT.reverse.bedgraph</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain" style="display:inline-block"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"># Output files:</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">#  - /path/to/S_output-basename_s5e20F1.gff</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">#  - /path/to/D_output-basename_s5e20F1.txt</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">#  - /path/to/O_output-basename_s5e20F1.txt</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">#  - /path/to/P_output-basename_s5e20F1.pdf</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">#  - /path/to/F_output-basename_s5e20F1.pdf</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">#  - /path/to/my-track-basename_forward.bw (optional)</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">#  - /path/to/my-track-basename_reverse.bw (optional)</span><br></span></code></pre><div class="buttonGroup__atx"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_eSgA" aria-hidden="true"><svg class="copyButtonIcon_y97N" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_LjdS" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div></div><footer class="theme-doc-footer docusaurus-mt-lg"><div class="theme-doc-footer-edit-meta-row row"><div class="col"></div><div class="col lastUpdated_vwxv"><span class="theme-last-updated">Last updated<!-- --> on <b><time datetime="2024-09-05T19:46:15.000Z">Sep 5, 2024</time></b> by <b>Olivia Wen-Mei Lang</b></span></div></div></footer></article><nav class="pagination-nav docusaurus-mt-lg" aria-label="Docs pages navigation"><a class="pagination-nav__link pagination-nav__link--prev" href="/labtutorial/Software/Websites/new-docs-website"><div class="pagination-nav__sublabel">Previous</div><div class="pagination-nav__label">New Website (docs)</div></a><a class="pagination-nav__link pagination-nav__link--next" href="/labtutorial/Software/UsingJetstream2/"><div class="pagination-nav__sublabel">Next</div><div class="pagination-nav__label">Using Jetstream2</div></a></nav></div></div><div class="col col--3"><div class="tableOfContents_bqdL thin-scrollbar theme-doc-toc-desktop"><ul class="table-of-contents table-of-contents__left-border"><li><a href="#download-scriptmanager-v014" class="table-of-contents__link toc-highlight">Download ScriptManager (v0.14):</a></li><li><a href="#download-python-scripts" class="table-of-contents__link toc-highlight">Download Python Scripts</a></li><li><a href="#optional-download-uscs-binary-bedgraphtobigwig" class="table-of-contents__link toc-highlight">(Optional) Download USCS binary <code>bedGraphToBigWig</code></a></li><li><a href="#download-data" class="table-of-contents__link toc-highlight">Download Data</a><ul><li><a href="#bam-file" class="table-of-contents__link toc-highlight">BAM File</a></li><li><a href="#optional-xxxxchromsizes-reference-file" class="table-of-contents__link toc-highlight">(Optional) XXXX.chrom.sizes Reference File</a></li></ul></li><li><a href="#convert-bam-file-to-scidx-format" class="table-of-contents__link toc-highlight">Convert BAM File to scIDX Format</a><ul><li><a href="#1-open-scriptmanager" class="table-of-contents__link toc-highlight">1. Open ScriptManager</a></li><li><a href="#2-generate-bai-index-file" class="table-of-contents__link toc-highlight">2. Generate BAI index file</a></li><li><a href="#3-create-a-whole-genome-pileup-using-bam-format-converter-tools" class="table-of-contents__link toc-highlight">3. Create a whole-genome pileup using BAM Format Converter Tools</a></li></ul></li><li><a href="#genetrack-script" class="table-of-contents__link toc-highlight">GeneTrack Script</a><ul><li><a href="#output-options" class="table-of-contents__link toc-highlight">Output Options</a></li></ul></li><li><a href="#cwpair-script" class="table-of-contents__link toc-highlight">CWPair Script</a><ul><li><a href="#output-options-1" class="table-of-contents__link toc-highlight">Output Options</a></li></ul></li><li><a href="#optional-view-pileup-in-genome-browser" class="table-of-contents__link toc-highlight">(Optional) View Pileup in Genome Browser</a><ul><li><a href="#convert-scidx-to-bigwig" class="table-of-contents__link toc-highlight">Convert scIDX to BigWig</a></li></ul></li><li><a href="#command-line-shell-script" class="table-of-contents__link toc-highlight">Command-Line shell script</a></li></ul></div></div></div></div></main></div></div><footer class="footer footer--dark"><div class="container container-fluid"><div class="row footer__links"><div class="col footer__col"><div class="footer__title">Docs</div><ul class="footer__items clean-list"><li class="footer__item"><a class="footer__link-item" href="/labtutorial/General">Wiki Home</a></li></ul></div><div class="col footer__col"><div class="footer__title">Community</div><ul class="footer__items clean-list"><li class="footer__item"><a href="https://pughlab.mbg.cornell.edu" target="_blank" rel="noopener noreferrer" class="footer__link-item">Pugh Lab<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_nPIU"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></a></li><li class="footer__item"><a href="https://williamkmlai.github.io" target="_blank" rel="noopener noreferrer" class="footer__link-item">Lai Lab<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_nPIU"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></a></li></ul></div><div class="col footer__col"><div class="footer__title">Other Tools We Develop</div><ul class="footer__items clean-list"><li class="footer__item"><a href="http://pughlab.mbg.cornell.edu/scriptmanager-docs" target="_blank" rel="noopener noreferrer" class="footer__link-item">ScriptManager<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_nPIU"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></a></li><li class="footer__item"><a href="https://github.com/seqcode/pegr" target="_blank" rel="noopener noreferrer" class="footer__link-item">PEGR<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_nPIU"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></a></li><li class="footer__item"><a href="http://pughlab.mbg.cornell.edu/stencil" target="_blank" rel="noopener noreferrer" class="footer__link-item">STENCIL<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_nPIU"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></a></li><li class="footer__item"><a href="http://pughlab.mbg.cornell.edu/GenoPipe-docs" target="_blank" rel="noopener noreferrer" class="footer__link-item">GenoPipe<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_nPIU"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></a></li></ul></div></div><div class="footer__bottom text--center"><div class="footer__copyright">Copyright © 2024 labtutorial. Built with Docusaurus.</div></div></div></footer></div>
<script src="/labtutorial/assets/js/runtime~main.bf2c6d3b.js"></script>
<script src="/labtutorial/assets/js/main.928c6cc0.js"></script>
</body>
</html>